---
title: "Differential gene expression analysis and larva weight analysis and plots"
author: "Vienna"
date: "4/28/2021"
output: html_document
  pdf_document: default
  keep_md: true
  html_document: default
self_contained: no
thumbnails: no
keep_md: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(warning = FALSE)
knitr::opts_chunk$set(message = FALSE)
options(stringsAsFactors = TRUE)

library(tidyverse)
library(sleuth)
library(readxl)
library(ggsignif)
library(kableExtra)
library(DESeq2)
library(tximport)
library(apeglm)
library(reshape)
library(plyr)
library(ggpubr)
```
# Larva weight experimental data

#### after defecating larva weight was taken of randomly chosen larva for each treatment (12-15 individuals per treatment)

```{r larva_weight}

weight<- read.table("larva_weight/Larva_weight.txt", header=TRUE)

my_comparisons <- list( c("Hive", "AG"), c("Hive", "C"), c("Hive", "BB"), c("Hive", "LG"), c("Hive", "LGBB"))

p<-ggerrorplot(weight, x = "treatment", y = "weight",desc_stat = "mean_se",error.plot = "errorbar",add ="mean",size=1.1,width=0.4,color = "gray35")+theme(legend.position = "none")+theme(legend.position = "none")+theme(axis.title.x = element_blank())+ ylab("Larval weight mg")
p<-p+ theme(axis.text.x = element_text(face="bold", size=14))+ theme(axis.text.y = element_text(face="bold", size=11))+theme(axis.title.y = element_text(face="bold", size=14))
p<-p+ geom_jitter(position=position_jitter(0.1), alpha=0.7,size=2.1,aes(colour=treatment))+ scale_color_brewer(palette="Dark2")
ggsave("larva_weight/larva_weight.png", height = 3.5, width = 7)

AG<-subset(weight, treatment == "AG")
BB<-subset(weight, treatment == "BB")
C<-subset(weight, treatment == "C")
LG<-subset(weight, treatment == "LG")
LGBB<-subset(weight, treatment == "LGBB")
Hive<-subset(weight, treatment == "Hive")

wilcox.test(AG$weight,Hive$weight)
wilcox.test(BB$weight,Hive$weight)
wilcox.test(C$weight,Hive$weight)
wilcox.test(LG$weight,Hive$weight)
wilcox.test(LGBB$weight,Hive$weight)

pvalues <- c(0.005816,0.01556,0.1088,0.002833,0.009368)
p.adjust(pvalues,method="fdr")

#[1] 0.01454000 0.01945000 0.10880000 0.01416500 0.01561333
```


# Differential gene expression analysis

#### mRNA sequencing was performed on the RNA of larvae samples (Qiagen Power Fecal RNA/DNA extraction Kit) and on RNA of adult whole abdomen (TRIzol extraction). Novaseq paired-end sequencing

```{r}
library(GenomicFeatures)
library(biomaRt)
library("systemPipeR")

mart <- useMart(biomart="metazoa_mart", host="metazoa.ensembl.org")
#datasets <- listDatasets(mart)
#datasets[1:40 , 1:2]

## choose my data set of interest

txdb <- makeTxDbFromBiomart(biomart="metazoa_mart",dataset="amellifera_eg_gene", host="metazoa.ensembl.org")
k <- keys(txdb, keytype = "TXNAME")
tx2gene <- select(txdb, k, "GENEID", "TXNAME")

#write_csv(tx2gene,"data_files/tx2gene_new.csv")
```

```{r load_data}
tx2gene <- read_csv("RNA/data_files/tx2gene_new.csv")

dirs <- list.files(file.path("RNA/data_files/kallisto/"))
quant_files <- paste0("RNA/data_files/kallisto/",dirs,"/abundance.tsv")
names(quant_files) <- dirs

txi <- tximport(quant_files,type="kallisto",tx2gene = tx2gene,ignoreTxVersion = TRUE)

sampleinfo <- read.delim("RNA/data_files/metafile_for_RNA_all.txt")
rownames(sampleinfo) <- sampleinfo$run

dds <- DESeqDataSetFromTximport(txi, colData = sampleinfo,design <- ~batch+condition)
#head(txi,2)
#all(rownames(dds) == colnames(txi))

#While it is not necessary to pre-filter low count genes before running the DESeq2 functions, it is useful: by removing rows in which there are very few reads, we reduce the memory size of the dds data object, and increase the speed of the transformation and testing functions within DESeq2.

keep <- rowSums(counts(dds)) > 5
dds <- dds[keep,]
```


### PCA of whole data set
```{r pca_all}

#first normalize with vst to produce log2 scale data
vstcounts <- vst(dds, blind=TRUE)
pcaData<-plotPCA(vstcounts, intgroup=c("life_state", "condition"),returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))

pcaData$condition<-factor(pcaData$condition,levels=c("hive_control","lab_food_nothing","adult_gut_transfer","bee_bread_addition","larva_gut_transfer","larva_gut_bee_bread"), labels=c("Hive","C","AG","BB","LG","LGBB"))
levels(pcaData$condition)

ggplot(pcaData, aes(PC1, PC2, color=condition, shape=life_state)) +
geom_point(size=4)+xlab(paste0("PC1: ",percentVar[1],"% variance")) +
ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
coord_fixed()+theme_bw()+theme(legend.title = element_blank())+ scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"))+ theme(legend.text = element_text(face="bold", size=15))+theme(axis.text = element_text(size=15))+theme(axis.title = element_text(size=16))+ylim(-50,50)+xlim(-75,75)

ggsave("RNA/pca_all.png", height = 6.5, width = 6.5)
```


## Larva samples 


```{r load_data}
tx2gene <- read_csv("RNA/data_files/tx2gene_new.csv")

#then import the kallisto files:

dirs <- list.files(file.path("RNA/data_files/kallisto_larva/"))
quant_files <- paste0("RNA/data_files/kallisto_larva/",dirs,"/abundance.tsv")
names(quant_files) <- dirs

txi <- tximport(quant_files,type="kallisto",tx2gene = tx2gene,ignoreTxVersion = TRUE)

sampleinfo <- read.delim("RNA/data_files/metafile_for_RNA_larva.txt")
rownames(sampleinfo) <- sampleinfo$run

dds_larva <- DESeqDataSetFromTximport(txi, colData = sampleinfo,design <- ~batch+condition)

keep <- rowSums(counts(dds)) > 5
dds <- dds[keep,]

#set a standard to be compared to (control)
dds_larva$condition <- relevel(dds_larva$condition, ref = "hive_control")

dds_larva <- DESeq(dds_larva)
```


### plot PCA larvae

```{r pca_larva}
vstcounts_l <- vst(dds_larva, blind=TRUE)
pcaData1<-plotPCA(vstcounts_l, intgroup="condition",returnData=TRUE)
percentVar1 <- round(100 * attr(pcaData1, "percentVar"))

pcaData1$condition<-factor(pcaData1$condition,levels=c("hive_control","lab_food_nothing","adult_gut_transfer","bee_bread_addition","larva_gut_transfer","larva_gut_bee_bread"), labels=c("Hive","C","AG","BB","LG","LGBB"))
levels(pcaData1$condition)

ggplot(pcaData1, aes(PC1, PC2, color=condition)) +
geom_point(size=4)+xlab(paste0("PC1: ",percentVar[1],"% variance")) +
ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
coord_fixed()+theme_bw()+theme(legend.title = element_blank())+ scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"))+ theme(legend.text = element_text(face="bold", size=15))+theme(axis.text = element_text(size=15))+theme(axis.title = element_text(size=16))+ylim(-50,50)+xlim(-75,75)

ggsave("RNA/Larvae-output/pca_larva.png", height = 5, width = 5)
```

### MA plots treatments vs hive control

```{r MA_plots}

#Shrinkage of effect size (LFC estimates) for visualization and ranking of genes.

resultsNames(dds_larva)

resLFC_larva_AG <- lfcShrink(dds_larva, coef="condition_adult_gut_transfer_vs_hive_control", type="apeglm")
resLFC_larva_BB <- lfcShrink(dds_larva, coef="condition_bee_bread_addition_vs_hive_control", type="apeglm")
resLFC_larva_C <- lfcShrink(dds_larva, coef="condition_lab_food_nothing_vs_hive_control", type="apeglm")
resLFC_larva_LGBB <- lfcShrink(dds_larva, coef="condition_larva_gut_bee_bread_vs_hive_control", type="apeglm")
resLFC_larva_LG <- lfcShrink(dds_larva, coef="condition_larva_gut_transfer_vs_hive_control", type="apeglm")
```

#### make MA graph combined with one legend
```{r, MA_figure_1}
library(ggpubr)

AG<-ggmaplot(resLFC_larva_AG, fdr = 0.05, fc = 1, size = 1.2,palette = c("#B31B21", "#1465AC", "darkgray"),genenames = as.vector(resLFC_larva_AG$name),top = 0,legend="top",label.select = NULL)+coord_cartesian(xlim = c(0, 18))+ scale_y_continuous(limits=c(-8, 8))+ theme(axis.text.x = element_text(size=16),axis.text.y = element_text(size=15),axis.title.x = element_text(size=17),axis.title.y = element_text(size=17),axis.line = element_line(size = 1, colour="gray20"),axis.ticks = element_line(size = 1, colour="gray20"))+guides(color = guide_legend(override.aes = list(size = c(3,3,3))))+theme(legend.position = c(0.70, 0.12),legend.text=element_text(size=14,face="bold"),legend.background = element_rect(fill="transparent"))+theme(plot.title = element_text(size=18, colour="gray30", face="bold",hjust=0.06, vjust=-5))+labs(title="Adult gut addition")

BB<-ggmaplot(resLFC_larva_BB, fdr = 0.05, fc = 1, size = 1.2,palette = c("#B31B21", "#1465AC", "darkgray"),genenames = as.vector(resLFC_larva_BB$name),top = 0)+coord_cartesian(xlim = c(0, 18))+ scale_y_continuous(limits=c(-8, 8))+ theme(axis.text.x = element_text(size=16),axis.text.y = element_text(size=15),axis.title.x = element_text(size=17),axis.title.y = element_text(size=17),axis.line = element_line(size = 1, colour="gray20"),axis.ticks = element_line(size = 1, colour="gray20"))+guides(color = guide_legend(override.aes = list(size = c(3,3,3))))+theme(legend.position = c(0.70, 0.12),legend.text=element_text(size=14,face = "bold"),legend.background = element_rect(fill="transparent"))+theme(axis.line.y = element_line(color = "white"))+ theme(axis.title.y = element_blank(),axis.text.y = element_blank(),axis.ticks.y = element_blank(),axis.title.x = element_blank(),axis.text.x = element_blank())+theme(plot.title = element_text(size=18, colour="gray30", face="bold",hjust=0.06, vjust=-5))+labs(title="Bee bread addition")

C<-ggmaplot(resLFC_larva_C, fdr = 0.05, fc = 1, size = 1.2,palette = c("#B31B21", "#1465AC", "darkgray"),genenames = as.vector(resLFC_larva_C$name),top = 0)+coord_cartesian(xlim = c(0, 18))+ scale_y_continuous(limits=c(-8, 8))+ theme(axis.text.x = element_text(size=16),axis.text.y = element_text(size=15),axis.title.x = element_text(size=17),axis.title.y = element_text(size=17),axis.line = element_line(size = 1, colour="gray20"),axis.ticks = element_line(size = 1, colour="gray20"))+guides(color = guide_legend(override.aes = list(size = c(3,3,3))))+theme(legend.position = c(0.70, 0.12),legend.text=element_text(size=14,face = "bold"),legend.background = element_rect(fill="transparent"))+theme(axis.line.y = element_line(color = "white"))+ theme(axis.title.y = element_blank(),axis.text.y = element_blank(),axis.ticks.y = element_blank(),axis.title.x = element_blank(),axis.text.x = element_blank())+theme(plot.title = element_text(size=18, colour="gray30", face="bold",hjust=0.06, vjust=-5))+labs(title="No addition")

LG<-ggmaplot(resLFC_larva_LG, fdr = 0.05, fc = 1, size = 1.2,palette = c("#B31B21", "#1465AC", "darkgray"),genenames = as.vector(resLFC_larva_LG$name),top = 0)+coord_cartesian(xlim = c(0, 18))+ scale_y_continuous(limits=c(-8, 8))+ theme(axis.text.x = element_text(size=16),axis.text.y = element_text(size=15),axis.title.x = element_text(size=17),axis.title.y = element_text(size=17),axis.line = element_line(size = 1, colour="gray20"),axis.ticks = element_line(size = 1, colour="gray20"))+guides(color = guide_legend(override.aes = list(size = c(3,3,3))))+theme(legend.position = c(0.70, 0.12),legend.text=element_text(size=14,face = "bold"),legend.background = element_rect(fill="transparent"))+theme(axis.line.y = element_line(color = "white"))+ theme(axis.title.y = element_blank(),axis.text.y = element_blank(),axis.ticks.y = element_blank(),axis.title.x = element_blank(),axis.text.x = element_blank())+theme(plot.title = element_text(size=18, colour="gray30", face="bold",hjust=0.06, vjust=-5))+labs(title="Larva gut addition")
  
LGBB<-ggmaplot(resLFC_larva_LGBB, fdr = 0.05, fc = 1, size = 1.2,palette = c("#B31B21", "#1465AC", "darkgray"),genenames = as.vector(resLFC_larva_LGBB$name),top = 0)+coord_cartesian(xlim = c(0, 18))+ scale_y_continuous(limits=c(-8, 8))+ theme(axis.text.x = element_text(size=16),axis.text.y = element_text(size=15),axis.title.x = element_text(size=17),axis.title.y = element_text(size=17),axis.line = element_line(size = 1, colour="gray20"),axis.ticks = element_line(size = 1, colour="gray20"))+guides(color = guide_legend(override.aes = list(size = c(3,3,3))))+theme(legend.position = c(0.70, 0.12),legend.text=element_text(size=14,face = "bold"),legend.background = element_rect(fill="transparent"))+theme(axis.line.y = element_line(color = "white"))+ theme(axis.title.y = element_blank(),axis.text.y = element_blank(),axis.ticks.y = element_blank(),axis.title.x = element_blank(),axis.text.x = element_blank())+theme(plot.title = element_text(size=18, colour="gray30", face="bold",hjust=0.06, vjust=-5))+labs(title="Larva gut & bee bread")


library(patchwork)
List <- list(AG,BB,C,LG,LGBB)
Plot <- wrap_plots(List,nrow=1)
ggsave("RNA/Larvae-output/larva_MA_all.png", height = 5, width = 15)
```


### extract significant differently expressed genes
#### set hive treatment as control everything is compared to

```{r extract sig genes}
#extract significant differently expressed genes
#get annotation for signicificant genes use biomart

### make the gene annotation list for Apis mellifera

#library("biomaRt")
#library("systemPipeR")

#mart <- useDataset('amellifera_eg_gene', mart)
#listAttributes(mart)
#functions <- getBM(attributes=c("ensembl_gene_id", "description"), mart=mart)
#functions <- functions[functions[,2]!="",]; functions[,2] <- as.character(functions[,2])

#write.table(functions, "data_files/annotationsBiomart.txt", quote=FALSE, row.names=FALSE, col.names=T, sep="\t")

### open and save as csv

functions <- read_csv("RNA/data_files/annotationsBiomart.csv", col_types = cols()) %>% dplyr::select(Gene_ID=ensembl_gene_id,description=description)

res_LG_sig <- results(dds_larva, tidy=TRUE, contrast=c("condition", "larva_gut_transfer", "hive_control"))
#write.csv(as.data.frame(res_LG_sig),file="RNA/Larvae-output/res_LG_all.csv")
res_LG_sig<- subset(res_LG_sig, padj < 0.05,pAdjustMethod = "fdr")
colnames(res_LG_sig)[which(names(res_LG_sig) == "row")] <- "Gene_ID"
res_LG_sig<-merge(res_LG_sig, functions, by="Gene_ID", type="left", match="first")
#write.csv(as.data.frame(res_LG_sig),file="RNA/Larvae-output/res_LG_sig.csv")

#How many adjusted p-values were less than 0.05?
sum(res_LG_sig$padj < 0.05, na.rm=TRUE)

res_AG_sig <- results(dds_larva, tidy=TRUE, contrast=c("condition", "adult_gut_transfer", "hive_control"))
#write.csv(as.data.frame(res_AG_sig),file="RNA/Larvae-output/res_AG_all.csv")
res_AG_sig<- subset(res_AG_sig, padj < 0.05,pAdjustMethod = "fdr")
colnames(res_AG_sig)[which(names(res_AG_sig) == "row")] <- "Gene_ID"
res_AG_sig<-merge(res_AG_sig, functions, by="Gene_ID", type="left", match="first")
#write.csv(as.data.frame(res_AG_sig),file="RNA/Larvae-output/res_AG_sig.csv")
sum(res_AG_sig$padj < 0.05, na.rm=TRUE)

res_C_sig <- results(dds_larva, tidy=TRUE, contrast=c("condition", "lab_food_nothing", "hive_control"))
#write.csv(as.data.frame(res_C_sig), file="RNA/Larvae-output/res_C_all.csv")
res_C_sig<- subset(res_C_sig, padj < 0.05)
colnames(res_C_sig)[which(names(res_C_sig) == "row")] <- "Gene_ID"
res_C_sig<-merge(res_C_sig, functions, by="Gene_ID", type="left", match="first")
#write.csv(as.data.frame(res_C_sig), file="RNA/Larvae-output/res_C_sig.csv")
sum(res_C_sig$padj < 0.05, na.rm=TRUE)

res_LGBB_sig <- results(dds_larva, tidy=TRUE, contrast=c("condition", "larva_gut_bee_bread", "hive_control"))
#write.csv(as.data.frame(res_LGBB_sig),file="RNA/Larvae-output/res_LGBB_all.csv")
res_LGBB_sig<- subset(res_LGBB_sig, padj < 0.05)
colnames(res_LGBB_sig)[which(names(res_LGBB_sig) == "row")] <- "Gene_ID"
res_LGBB_sig<-merge(res_LGBB_sig, functions, by="Gene_ID", type="left", match="first")
#write.csv(as.data.frame(res_LGBB_sig), file="RNA/Larvae-output/res_LGBB_sig.csv")
sum(res_LGBB_sig$padj < 0.05, na.rm=TRUE)

res_BB_sig <- results(dds_larva, tidy=TRUE, contrast=c("condition", "bee_bread_addition", "hive_control"))
#write.csv(as.data.frame(res_BB_sig), file="RNA/Larvae-output/res_BB_all.csv")
res_BB_sig<- subset(res_BB_sig, padj < 0.05)
colnames(res_BB_sig)[which(names(res_BB_sig) == "row")] <- "Gene_ID"
res_BB_sig<-merge(res_BB_sig, functions, by="Gene_ID", type="left", match="first")
#write.csv(as.data.frame(res_BB_sig), file="RNA/Larvae-output/res_BB_sig.csv")
sum(res_BB_sig$padj < 0.05, na.rm=TRUE)
```

#### Go-term annotation

```{r GO_larvae}

library(AnnotationDbi)
library(GSEABase)
library(Category)
library(GOstats)

### make the go list for Apis mellifera

#library("biomaRt")
#library("systemPipeR")

#mart <- useDataset('amellifera_eg_gene', mart)
#listAttributes(mart)
#go <- getBM(attributes=c("ensembl_gene_id","go_id", "description","name_1006", "namespace_1003"), mart=mart)
#go <- go[go[,5]!="",]; go[,5] <- as.character(go[,5])

#go[go[,3]=="molecular_function", 3] <- "F"; go[go[,3]=="biological_process", 3] <- "P"; go[go[,3]=="cellular_component", 3] <- "C"
#go[1:4,]

#dir.create("data_files/GO")
#write.table(go, "data_files/GO/GOannotationsBiomart_mod.txt", quote=FALSE, row.names=FALSE, col.names=T, sep="\t")
#catdb <- makeCATdb(myfile="data_files/GO/GOannotationsBiomart_mod.txt", lib=NULL, org="", colno=c(1,2,3,4), idconv=NULL)
#save(catdb, file="data_files/GO/catdb.RData")

### open and save as csv

go <- read_csv("RNA/data_files/GO/GOannotationsBiomart_mod.csv", col_types = cols()) %>% mutate(evidence = "ISS") %>% dplyr::select(GO=go_id, evidence,gene=ensembl_gene_id)
universe <- unique(go$gene)
goFrame <- GOFrame(as.data.frame(go %>% dplyr::filter(gene %in% universe)), organism="Apis mellifera")
goAllFrame <- GOAllFrame(goFrame)
gsc <- GeneSetCollection(goAllFrame, setType = GOCollection())
```

All lists with significant expressed genes are split into up- and down-regulated to receive individual Go-term files for each

```{r GO_AG}
#up
AG_up <- read.csv("RNA/Larvae-output/res_AG_sig_up.csv")

DsxModuleGo <- hyperGTest(GSEAGOHyperGParams(name = "enriched in AG treatment",
    geneSetCollection=gsc,geneIds = intersect(AG_up$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))

summary(DsxModuleGo) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 

GO_enrich.AG <- as.data.frame(summary(DsxModuleGo))
GO_enrich.AG %>% 
arrange(Pvalue) %>% 
#write.csv(file = "RNA/Larvae-output/Go/GO_term_enrichment_AG_up.csv")

### make list
hub_genesGO <- read.csv("RNA/Larvae-output/Go/GO_term_enrichment_AG_up.csv")
# keep only highly significant terms
hub_genesGO <- hub_genesGO %>%
filter(Pvalue < 0.01) 

#plot it
g <- hub_genesGO %>%
  mutate(logPv = log(Pvalue))
ggplot(g, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("AG up compared to Hive")+theme(axis.title.y = element_blank())
ggsave("RNA/Larvae-output/Go/AG_up.png", height = 8, width = 9)

#down
AG_down <- read.csv("RNA/Larvae-output/res_AG_sig_down.csv")

DsxModuleGo2 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in AG treatment",
    geneSetCollection=gsc,geneIds = intersect(AG_down$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))
summary(DsxModuleGo2) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 
GO_enrich.AG2 <- as.data.frame(summary(DsxModuleGo2))
GO_enrich.AG2 %>% 
arrange(Pvalue) %>% 
#write.csv(file = "RNA/Larvae-output/Go/GO_term_enrichment_AG_down.csv")
hub_genesGO2 <- read.csv("Larvae-output/Go/GO_term_enrichment_AG_down.csv")
# keep only highly significant terms
hub_genesGO2 <- hub_genesGO2 %>%
filter(Pvalue < 0.01) 

g2 <- hub_genesGO2 %>%
  mutate(logPv = log(Pvalue))
#or plot it based on the count of each term and the pvalue:
ggplot(g2, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("AG down compared to Hive")+theme(axis.title.y = element_blank())
ggsave("RNA/Larvae-output/Go/AG_down.png", height = 8, width = 9)
```


```{r GO_BB}

#up
BB_up <- read.csv("Larvae-output/res_BB_sig_up.csv")

DsxModuleGo3 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in BB treatment",
    geneSetCollection=gsc,geneIds = intersect(BB_up$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))

summary(DsxModuleGo3) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 

GO_enrich.BB <- as.data.frame(summary(DsxModuleGo3))
GO_enrich.BB %>% 
arrange(Pvalue) %>% 
write.csv(file = "Larvae-output/Go/GO_term_enrichment_BB_up.csv")

hub_genesGO <- read.csv("Larvae-output/Go/GO_term_enrichment_BB_up.csv")
# keep only highly significant terms
hub_genesGO <- hub_genesGO %>%
filter(Pvalue < 0.01) 
g <- hub_genesGO %>%
  mutate(logPv = log(Pvalue))
ggplot(g, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("BB up compared to Hive")+theme(axis.title.y = element_blank())
ggsave("Larvae-output/Go/BB_up.png", height = 8, width = 9)

#down
BB_down <- read.csv("Larvae-output/res_BB_sig_down.csv")

DsxModuleGo2 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in BB treatment",
    geneSetCollection=gsc,geneIds = intersect(BB_down$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))
summary(DsxModuleGo2) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 
GO_enrich.BB2 <- as.data.frame(summary(DsxModuleGo2))
GO_enrich.BB2 %>% 
arrange(Pvalue) %>% 
write.csv(file = "Larvae-output/Go/GO_term_enrichment_BB_down.csv")
hub_genesGO2 <- read.csv("Larvae-output/Go/GO_term_enrichment_BB_down.csv")
# keep only highly significant terms
hub_genesGO2 <- hub_genesGO2 %>%
filter(Pvalue < 0.01) 

g2 <- hub_genesGO2 %>%
  mutate(logPv = log(Pvalue))
ggplot(g2, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("BB down compared to Hive")+theme(axis.title.y = element_blank())
ggsave("Larvae-output/Go/BB_down.png", height = 8, width = 9)
```



```{r GO_C}

#up
C_up <- read.csv("Larvae-output/res_C_sig_up.csv")

DsxModuleGo4 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in C treatment",
    geneSetCollection=gsc,geneIds = intersect(C_up$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))

summary(DsxModuleGo4) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 

GO_enrich.C <- as.data.frame(summary(DsxModuleGo4))
GO_enrich.C %>% 
arrange(Pvalue) %>% 
write.csv(file = "Larvae-output/Go/GO_term_enrichment_C_up.csv")

hub_genesGO <- read.csv("Larvae-output/Go/GO_term_enrichment_C_up.csv")
# keep only highly significant terms
hub_genesGO <- hub_genesGO %>%
filter(Pvalue < 0.01) 
g <- hub_genesGO %>%
  mutate(logPv = log(Pvalue))
ggplot(g, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("C up compared to Hive")+theme(axis.title.y = element_blank())
ggsave("Larvae-output/Go/C_up.png", height = 8, width = 9)

#down
C_down <- read.csv("Larvae-output/res_C_sig_down.csv")

DsxModuleGo5 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in C treatment",
    geneSetCollection=gsc,geneIds = intersect(C_down$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))
summary(DsxModuleGo5) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 
GO_enrich.C2 <- as.data.frame(summary(DsxModuleGo5))
GO_enrich.C2 %>% 
arrange(Pvalue) %>% 
write.csv(file = "Larvae-output/Go/GO_term_enrichment_C_down.csv")
hub_genesGO2 <- read.csv("Larvae-output/Go/GO_term_enrichment_C_down.csv")
# keep only highly significant terms
hub_genesGO2 <- hub_genesGO2 %>%
filter(Pvalue < 0.01) 

g2 <- hub_genesGO2 %>%
  mutate(logPv = log(Pvalue))
ggplot(g2, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("C down compared to Hive")+theme(axis.title.y = element_blank())
ggsave("Larvae-output/Go/C_down.png", height = 8, width = 9)
```


```{r GO_LG}

#up
LG_up <- read.csv("Larvae-output/res_LG_sig_up.csv")

DsxModuleGo6 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in LG treatment",
    geneSetCollection=gsc,geneIds = intersect(LG_up$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))

summary(DsxModuleGo6) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 

GO_enrich.LG <- as.data.frame(summary(DsxModuleGo6))
GO_enrich.LG %>% 
arrange(Pvalue) %>% 
write.csv(file = "Larvae-output/Go/GO_term_enrichment_LG_up.csv")

hub_genesGO <- read.csv("Larvae-output/Go/GO_term_enrichment_LG_up.csv")
# keep only highly significant terms
hub_genesGO <- hub_genesGO %>%
filter(Pvalue < 0.01) 
g <- hub_genesGO %>%
  mutate(logPv = log(Pvalue))
ggplot(g, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("LG up compared to Hive")+theme(axis.title.y = element_blank())
ggsave("Larvae-output/Go/LG_up.png", height = 8, width = 9)

#down
LG_down <- read.csv("Larvae-output/res_LG_sig_down.csv")

DsxModuleGo7 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in LG treatment",
    geneSetCollection=gsc,geneIds = intersect(LG_down$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))
summary(DsxModuleGo7) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 
GO_enrich.LG2 <- as.data.frame(summary(DsxModuleGo7))
GO_enrich.LG2 %>% 
arrange(Pvalue) %>% 
write.csv(file = "Larvae-output/Go/GO_term_enrichment_LG_down.csv")
hub_genesGO2 <- read.csv("Larvae-output/Go/GO_term_enrichment_LG_down.csv")
# keep only highly significant terms
hub_genesGO2 <- hub_genesGO2 %>%
filter(Pvalue < 0.01) 

g2 <- hub_genesGO2 %>%
  mutate(logPv = log(Pvalue))
ggplot(g2, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("LG down compared to Hive")+theme(axis.title.y = element_blank())
ggsave("Larvae-output/Go/LG_down.png", height = 8, width = 9)
```


```{r GO_LGBB}

#up
LGBB_up <- read.csv("Larvae-output/res_LGBB_sig_up.csv")

DsxModuleGo8 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in LGBB treatment",
    geneSetCollection=gsc,geneIds = intersect(LGBB_up$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))

summary(DsxModuleGo8) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 

GO_enrich.LGBB <- as.data.frame(summary(DsxModuleGo8))
GO_enrich.LGBB %>% 
arrange(Pvalue) %>% 
write.csv(file = "Larvae-output/Go/GO_term_enrichment_LGBB_up.csv")

hub_genesGO <- read.csv("Larvae-output/Go/GO_term_enrichment_LGBB_up.csv")
# keep only highly significant terms
hub_genesGO <- hub_genesGO %>%
filter(Pvalue < 0.01) 
g <- hub_genesGO %>%
  mutate(logPv = log(Pvalue))
ggplot(g, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("LGBB up compared to Hive")+theme(axis.title.y = element_blank())
ggsave("Larvae-output/Go/LGBB_up.png", height = 8, width = 9)

#down
LGBB_down <- read.csv("Larvae-output/res_LGBB_sig_down.csv")

DsxModuleGo9 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in LGBB treatment",
    geneSetCollection=gsc,geneIds = intersect(LGBB_down$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))
summary(DsxModuleGo9) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 
GO_enrich.LGBB2 <- as.data.frame(summary(DsxModuleGo9))
GO_enrich.LGBB2 %>% 
arrange(Pvalue) %>% 
write.csv(file = "Larvae-output/Go/GO_term_enrichment_LGBB_down.csv")
hub_genesGO2 <- read.csv("Larvae-output/Go/GO_term_enrichment_LGBB_down.csv")
# keep only highly significant terms
hub_genesGO2 <- hub_genesGO2 %>%
filter(Pvalue < 0.01) 

g2 <- hub_genesGO2 %>%
  mutate(logPv = log(Pvalue))
ggplot(g2, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("LGBB down compared to Hive")+theme(axis.title.y = element_blank())
ggsave("Larvae-output/Go/LGBB_down.png", height = 8, width = 9)
```


### set comparison against C nothing-added lab treatment to be able to disentangle effects from diet components on lab-reared larva

```{r load_data2}
tx2gene <- read_csv("data_files/tx2gene_new.csv")

#then import the kallisto files:

dirs <- list.files(file.path("data_files/kallisto_larva/"))
quant_files <- paste0("data_files/kallisto_larva/",dirs,"/abundance.tsv")
names(quant_files) <- dirs

txi <- tximport(quant_files,type="kallisto",tx2gene = tx2gene,ignoreTxVersion = TRUE)

sampleinfo <- read.delim("data_files/metafile_for_RNA_larva.txt")
rownames(sampleinfo) <- sampleinfo$run

dds <- DESeqDataSetFromTximport(txi, colData = sampleinfo,design <- ~batch+condition)
#head(txi,2)
#all(rownames(dds) == colnames(txi))

#While it is not necessary to pre-filter low count genes before running the DESeq2 functions, it is useful: by removing rows in which there are very few reads, we reduce the memory size of the dds data object, and increase the speed of the transformation and testing functions within DESeq2.

keep <- rowSums(counts(dds)) > 5
dds <- dds[keep,]

#set a standard to be compared to (control)
dds_larva$condition <- relevel(dds_larva$condition, ref = "lab_food_nothing")

dds_larva <- DESeq(dds_larva)
```

#### make graph combined with one legend
```{r, figure_1}

resLFC_larva_AG <- lfcShrink(dds_larva, coef="condition_adult_gut_transfer_vs_lab_food_nothing", type="apeglm")
resLFC_larva_BB <- lfcShrink(dds_larva, coef="condition_bee_bread_addition_vs_lab_food_nothing" , type="apeglm")
resLFC_larva_Hive <- lfcShrink(dds_larva, coef="condition_hive_control_vs_lab_food_nothing" , type="apeglm")
resLFC_larva_LGBB <- lfcShrink(dds_larva, coef="condition_larva_gut_bee_bread_vs_lab_food_nothing", type="apeglm")
resLFC_larva_LG <- lfcShrink(dds_larva, coef="condition_larva_gut_transfer_vs_lab_food_nothing", type="apeglm")

AG<-ggmaplot(resLFC_larva_AG, fdr = 0.05, fc = 1, size = 1.2,palette = c("#B31B21", "#1465AC", "darkgray"),genenames = as.vector(resLFC_larva_AG$name),top = 0,legend="top",label.select = NULL)+coord_cartesian(xlim = c(0, 18))+ scale_y_continuous(limits=c(-6, 8))+ theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14),axis.title.x = element_text(size=15),axis.title.y = element_text(size=15),axis.line = element_line(size = 1, colour="gray20"),axis.ticks = element_line(size = 1, colour="gray20"))+guides(color = guide_legend(override.aes = list(size = c(3,3,3))))+theme(legend.position = c(0.14, 0.12),legend.text=element_text(size=13),legend.background = element_rect(fill="transparent"))+theme(plot.title = element_text(size=17, colour="gray30", face="bold",hjust=0.06, vjust=-5))+labs(title="Adult gut addition")

BB<-ggmaplot(resLFC_larva_BB, fdr = 0.05, fc = 1, size = 1.2,palette = c("#B31B21", "#1465AC", "darkgray"),genenames = as.vector(resLFC_larva_BB$name),top = 0)+coord_cartesian(xlim = c(0, 18))+ scale_y_continuous(limits=c(-6, 8))+ theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14),axis.title.x = element_text(size=15),axis.title.y = element_text(size=15),axis.line = element_line(size = 1, colour="gray20"),axis.ticks.y = element_line(size = 1, colour="gray20"))+guides(color = guide_legend(override.aes = list(size = c(3,3,3))))+theme(legend.position = c(0.13, 0.12),legend.text=element_text(size=13),legend.background = element_rect(fill="transparent"))+theme(axis.line.y = element_line(color = "white"))+ theme(axis.title.y = element_blank(),axis.text.y = element_blank(),axis.ticks.y = element_blank(),axis.title.x = element_blank(),axis.text.x = element_blank())+theme(plot.title = element_text(size=17, colour="gray30", face="bold",hjust=0.06, vjust=-5))+labs(title="Bee bread addition")

Hive<-ggmaplot(resLFC_larva_Hive, fdr = 0.05, fc = 1, size = 1.2,palette = c("#B31B21", "#1465AC", "darkgray"),genenames = as.vector(resLFC_larva_Hive$name),top = 0)+coord_cartesian(xlim = c(0, 18))+ scale_y_continuous(limits=c(-6, 8))+ theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14),axis.title.x = element_text(size=15),axis.title.y = element_text(size=15),axis.line = element_line(size = 1, colour="gray20"),axis.ticks = element_line(size = 1, colour="gray20"))+guides(color = guide_legend(override.aes = list(size = c(3,3,3))))+theme(legend.position = c(0.13, 0.12),legend.text=element_text(size=13),legend.background = element_rect(fill="transparent"))+theme(axis.line.y = element_line(color = "white"))+ theme(axis.title.y = element_blank(),axis.text.y = element_blank(),axis.ticks.y = element_blank(),axis.title.x = element_blank(),axis.text.x = element_blank())+theme(plot.title = element_text(size=17, colour="gray30", face="bold",hjust=0.06, vjust=-5))+labs(title="Hive control")

LG<-ggmaplot(resLFC_larva_LG, fdr = 0.05, fc = 1, size = 1.2,palette = c("#B31B21", "#1465AC", "darkgray"),genenames = as.vector(resLFC_larva_LG$name),top = 0,legend="top",label.select = NULL)+coord_cartesian(xlim = c(0, 18))+ scale_y_continuous(limits=c(-6, 8))+ theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14),axis.title.x = element_text(size=15),axis.title.y = element_text(size=15),axis.line = element_line(size = 1, colour="gray20"),axis.ticks = element_line(size = 1, colour="gray20"))+guides(color = guide_legend(override.aes = list(size = c(3,3,3))))+theme(legend.position = c(0.14, 0.12),legend.text=element_text(size=13),legend.background = element_rect(fill="transparent"))+theme(plot.title = element_text(size=17, colour="gray30", face="bold",hjust=0.06, vjust=-5))+labs(title="Larva gut addition")
  
LGBB<-ggmaplot(resLFC_larva_LGBB, fdr = 0.05, fc = 1, size = 1.2,palette = c("#B31B21", "#1465AC", "darkgray"),genenames = as.vector(resLFC_larva_LGBB$name),top = 0)+coord_cartesian(xlim = c(0, 18))+ scale_y_continuous(limits=c(-6, 8))+ theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14),axis.title.x = element_text(size=15),axis.title.y = element_text(size=15),axis.line = element_line(size = 1, colour="gray20"),axis.ticks = element_line(size = 1, colour="gray20"))+guides(color = guide_legend(override.aes = list(size = c(3,3,3))))+theme(legend.position = c(0.13, 0.12),legend.text=element_text(size=13),legend.background = element_rect(fill="transparent"))+theme(axis.line.y = element_line(color = "white"))+ theme(axis.title.y = element_blank(),axis.text.y = element_blank(),axis.ticks.y = element_blank(),axis.title.x = element_blank(),axis.text.x = element_blank())+theme(plot.title = element_text(size=17, colour="gray30", face="bold",hjust=0.06, vjust=-5))+labs(title="Larva gut & bee bread addition")
   
library(patchwork)
List <- list(AG,BB,Hive,LG,LGBB)
Plot <- wrap_plots(List)
ggsave("Larvae-output/compare_against_C_lab_treatment/larva_MA_against_lab_control.png", height = 10, width = 15)

```

```{r extract sig genes2}

functions <- read_csv("data_files/annotationsBiomart.csv", col_types = cols()) %>% dplyr::select(Gene_ID=ensembl_gene_id,description=description)

res_LG_sig <- results(dds_larva, tidy=TRUE, contrast=c("condition", "larva_gut_transfer", "lab_food_nothing"))
write.csv(as.data.frame(res_LG_sig), file="Larvae-output/compare_against_C_lab_treatment/res_LG_all.csv")
res_LG_sig<- subset(res_LG_sig, padj < 0.05,pAdjustMethod = "fdr")
colnames(res_LG_sig)[which(names(res_LG_sig) == "row")] <- "Gene_ID"
res_LG_sig<-merge(res_LG_sig, functions, by="Gene_ID", type="left", match="first")
write.csv(as.data.frame(res_LG_sig), file="Larvae-output/compare_against_C_lab_treatment/res_LG_sig.csv")

#How many adjusted p-values were less than 0.05?
sum(res_LG_sig$padj < 0.05, na.rm=TRUE)

res_AG_sig <- results(dds_larva, tidy=TRUE, contrast=c("condition", "adult_gut_transfer", "lab_food_nothing"))
write.csv(as.data.frame(res_AG_sig), file="Larvae-output/compare_against_C_lab_treatment/res_AG_all.csv")
res_AG_sig<- subset(res_AG_sig, padj < 0.05,pAdjustMethod = "fdr")
colnames(res_AG_sig)[which(names(res_AG_sig) == "row")] <- "Gene_ID"
res_AG_sig<-merge(res_AG_sig, functions, by="Gene_ID", type="left", match="first")
write.csv(as.data.frame(res_AG_sig), file="Larvae-output/compare_against_C_lab_treatment/res_AG_sig.csv")
sum(res_AG_sig$padj < 0.05, na.rm=TRUE)

res_LGBB_sig <- results(dds_larva, tidy=TRUE, contrast=c("condition", "larva_gut_bee_bread", "lab_food_nothing"))
write.csv(as.data.frame(res_LGBB_sig), file="Larvae-output/compare_against_C_lab_treatment/res_LGBB_all.csv")
res_LGBB_sig<- subset(res_LGBB_sig, padj < 0.05)
colnames(res_LGBB_sig)[which(names(res_LGBB_sig) == "row")] <- "Gene_ID"
res_LGBB_sig<-merge(res_LGBB_sig, functions, by="Gene_ID", type="left", match="first")
write.csv(as.data.frame(res_LGBB_sig), file="Larvae-output/compare_against_C_lab_treatment/res_LGBB_sig.csv")
sum(res_LGBB_sig$padj < 0.05, na.rm=TRUE)

res_BB_sig <- results(dds_larva, tidy=TRUE, contrast=c("condition", "bee_bread_addition", "lab_food_nothing"))
write.csv(as.data.frame(res_BB_sig), file="Larvae-output/compare_against_C_lab_treatment/res_BB_all.csv")
res_BB_sig<- subset(res_BB_sig, padj < 0.05)
colnames(res_BB_sig)[which(names(res_BB_sig) == "row")] <- "Gene_ID"
res_BB_sig<-merge(res_BB_sig, functions, by="Gene_ID", type="left", match="first")
write.csv(as.data.frame(res_BB_sig), file="Larvae-output/compare_against_C_lab_treatment/res_BB_sig.csv")
sum(res_BB_sig$padj < 0.05, na.rm=TRUE)
```

```{r GO_AG2}

#up
AG_up <- read.csv("Larvae-output/compare_against_C_lab_treatment/res_AG_sig_up.csv")

DsxModuleGo <- hyperGTest(GSEAGOHyperGParams(name = "enriched in AG treatment",
    geneSetCollection=gsc,geneIds = intersect(AG_up$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))

summary(DsxModuleGo) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 

GO_enrich.AG <- as.data.frame(summary(DsxModuleGo))
GO_enrich.AG %>% 
arrange(Pvalue) %>% 
write.csv(file = "Larvae-output/compare_against_C_lab_treatment/Go/GO_term_enrichment_AG_up.csv")

### make list
hub_genesGO <- read.csv("Larvae-output/compare_against_C_lab_treatment/Go/GO_term_enrichment_AG_up.csv")
# keep only highly significant terms
hub_genesGO <- hub_genesGO %>%
filter(Pvalue < 0.01) 

#plot it
g <- hub_genesGO %>%
  mutate(logPv = log(Pvalue))
ggplot(g, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("AG up compared to C")+theme(axis.title.y = element_blank())
ggsave("Larvae-output/compare_against_C_lab_treatment/Go/AG_up.png", height = 8, width = 9)

#down
AG_down <- read.csv("Larvae-output/compare_against_C_lab_treatment/res_AG_sig_down.csv")

DsxModuleGo2 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in AG treatment",
    geneSetCollection=gsc,geneIds = intersect(AG_down$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))
summary(DsxModuleGo2) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 
GO_enrich.AG2 <- as.data.frame(summary(DsxModuleGo2))
GO_enrich.AG2 %>% 
arrange(Pvalue) %>% 
write.csv(file = "Larvae-output/compare_against_C_lab_treatment/Go/GO_term_enrichment_AG_down.csv")
hub_genesGO2 <- read.csv("Larvae-output/compare_against_C_lab_treatment/Go/GO_term_enrichment_AG_down.csv")
# keep only highly significant terms
hub_genesGO2 <- hub_genesGO2 %>%
filter(Pvalue < 0.01) 

g2 <- hub_genesGO2 %>%
  mutate(logPv = log(Pvalue))
#or plot it based on the count of each term and the pvalue:
ggplot(g2, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("AG down compared to C")+theme(axis.title.y = element_blank())
ggsave("Larvae-output/compare_against_C_lab_treatment/Go/AG_down.png", height = 8, width = 9)
```


```{r GO_BB2}

#up
BB_up <- read.csv("Larvae-output/compare_against_C_lab_treatment/res_BB_sig_up.csv")

DsxModuleGo3 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in BB treatment",
    geneSetCollection=gsc,geneIds = intersect(BB_up$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))

summary(DsxModuleGo3) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 

GO_enrich.BB <- as.data.frame(summary(DsxModuleGo3))
GO_enrich.BB %>% 
arrange(Pvalue) %>% 
write.csv(file = "Larvae-output/compare_against_C_lab_treatment/Go/GO_term_enrichment_BB_up.csv")

hub_genesGO <- read.csv("Larvae-output/compare_against_C_lab_treatment/Go/GO_term_enrichment_BB_up.csv")
# keep only highly significant terms
hub_genesGO <- hub_genesGO %>%
filter(Pvalue < 0.01) 
g <- hub_genesGO %>%
  mutate(logPv = log(Pvalue))
ggplot(g, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("BB up compared to C")+theme(axis.title.y = element_blank())
ggsave("Larvae-output/compare_against_C_lab_treatment/Go/BB_up.png", height = 8, width = 9)

#down
BB_down <- read.csv("Larvae-output/compare_against_C_lab_treatment/res_BB_sig_down.csv")

DsxModuleGo2 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in BB treatment",
    geneSetCollection=gsc,geneIds = intersect(BB_down$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))
summary(DsxModuleGo2) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 
GO_enrich.BB2 <- as.data.frame(summary(DsxModuleGo2))
GO_enrich.BB2 %>% 
arrange(Pvalue) %>% 
write.csv(file = "Larvae-output/compare_against_C_lab_treatment/Go/GO_term_enrichment_BB_down.csv")
hub_genesGO2 <- read.csv("Larvae-output/compare_against_C_lab_treatment/Go/GO_term_enrichment_BB_down.csv")
# keep only highly significant terms
hub_genesGO2 <- hub_genesGO2 %>%
filter(Pvalue < 0.01) 

g2 <- hub_genesGO2 %>%
  mutate(logPv = log(Pvalue))
ggplot(g2, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("BB down compared to C")+theme(axis.title.y = element_blank())
ggsave("Larvae-output/compare_against_C_lab_treatment/Go/BB_down.png", height = 8, width = 9)
```

```{r GO_LG2}

#up
LG_up <- read.csv("Larvae-output/compare_against_C_lab_treatment/res_LG_sig_up.csv")

DsxModuleGo6 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in LG treatment",
    geneSetCollection=gsc,geneIds = intersect(LG_up$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))

summary(DsxModuleGo6) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 

GO_enrich.LG <- as.data.frame(summary(DsxModuleGo6))
GO_enrich.LG %>% 
arrange(Pvalue) %>% 
write.csv(file = "Larvae-output/compare_against_C_lab_treatment/Go/GO_term_enrichment_LG_up.csv")

hub_genesGO <- read.csv("Larvae-output/compare_against_C_lab_treatment/Go/GO_term_enrichment_LG_up.csv")
# keep only highly significant terms
hub_genesGO <- hub_genesGO %>%
filter(Pvalue < 0.01) 
g <- hub_genesGO %>%
  mutate(logPv = log(Pvalue))
ggplot(g, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("LG up compared to C")+theme(axis.title.y = element_blank())
ggsave("Larvae-output/compare_against_C_lab_treatment/Go/LG_up.png", height = 8, width = 9)

#down
LG_down <- read.csv("Larvae-output/compare_against_C_lab_treatment/res_LG_sig_down.csv")

DsxModuleGo7 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in LG treatment",
    geneSetCollection=gsc,geneIds = intersect(LG_down$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))
summary(DsxModuleGo7) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 
GO_enrich.LG2 <- as.data.frame(summary(DsxModuleGo7))
GO_enrich.LG2 %>% 
arrange(Pvalue) %>% 
write.csv(file = "Larvae-output/compare_against_C_lab_treatment/Go/GO_term_enrichment_LG_down.csv")
hub_genesGO2 <- read.csv("Larvae-output/compare_against_C_lab_treatment/Go/GO_term_enrichment_LG_down.csv")
# keep only highly significant terms
hub_genesGO2 <- hub_genesGO2 %>%
filter(Pvalue < 0.01) 

g2 <- hub_genesGO2 %>%
  mutate(logPv = log(Pvalue))
ggplot(g2, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("LG down compared to C")+theme(axis.title.y = element_blank())
ggsave("Larvae-output/compare_against_C_lab_treatment/Go/LG_down.png", height = 8, width = 9)
```


```{r GO_LGBB2}

#up
LGBB_up <- read.csv("Larvae-output/compare_against_C_lab_treatment/res_LGBB_sig_up.csv")

DsxModuleGo8 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in LGBB treatment",
    geneSetCollection=gsc,geneIds = intersect(LGBB_up$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))

summary(DsxModuleGo8) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 

GO_enrich.LGBB <- as.data.frame(summary(DsxModuleGo8))
GO_enrich.LGBB %>% 
arrange(Pvalue) %>% 
write.csv(file = "Larvae-output/compare_against_C_lab_treatment/Go/GO_term_enrichment_LGBB_up.csv")

hub_genesGO <- read.csv("Larvae-output/compare_against_C_lab_treatment/Go/GO_term_enrichment_LGBB_up.csv")
# keep only highly significant terms
hub_genesGO <- hub_genesGO %>%
filter(Pvalue < 0.01) 
g <- hub_genesGO %>%
  mutate(logPv = log(Pvalue))
ggplot(g, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("LGBB up compared to C")+theme(axis.title.y = element_blank())
ggsave("Larvae-output/compare_against_C_lab_treatment/Go/LGBB_up.png", height = 8, width = 9)

#down
LGBB_down <- read.csv("Larvae-output/compare_against_C_lab_treatment/res_LGBB_sig_down.csv")

DsxModuleGo9 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in LGBB treatment",
    geneSetCollection=gsc,geneIds = intersect(LGBB_down$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))
summary(DsxModuleGo9) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 
GO_enrich.LGBB2 <- as.data.frame(summary(DsxModuleGo9))
GO_enrich.LGBB2 %>% 
arrange(Pvalue) %>% 
write.csv(file = "Larvae-output/compare_against_C_lab_treatment/Go/GO_term_enrichment_LGBB_down.csv")
hub_genesGO2 <- read.csv("Larvae-output/compare_against_C_lab_treatment/Go/GO_term_enrichment_LGBB_down.csv")
# keep only highly significant terms
hub_genesGO2 <- hub_genesGO2 %>%
filter(Pvalue < 0.02) #<- set to 0.02 as below 0.01 nothing available

g2 <- hub_genesGO2 %>%
  mutate(logPv = log(Pvalue))
ggplot(g2, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("LGBB down compared to C")+theme(axis.title.y = element_blank())
ggsave("Larvae-output/compare_against_C_lab_treatment/Go/LGBB_down.png", height = 8, width = 9)
```


## Adults gene expression


```{r load_data}
tx2gene <- read_csv("data_files/tx2gene_new.csv")

#then import the kallisto files:

dirs <- list.files(file.path("data_files/kallisto_adults/"))
quant_files <- paste0("data_files/kallisto_adults/",dirs,"/abundance.tsv")
names(quant_files) <- dirs

txi <- tximport(quant_files,type="kallisto",tx2gene = tx2gene,ignoreTxVersion = TRUE)

sampleinfo <- read.delim("data_files/metafile_for_RNA_adults.txt")
rownames(sampleinfo) <- sampleinfo$run

dds <- DESeqDataSetFromTximport(txi, colData = sampleinfo,design <- ~condition)
keep <- rowSums(counts(dds)) > 5
dds <- dds[keep,]

#set a standard to be compared to (control)
dds_adults$condition <- relevel(dds_adults$condition, ref = "hive_control")

dds_adults <- DESeq(dds_adults)
```


### plot PCA adults

```{r pca_adults, fig.width=6, fig.height=5.5}
vstcounts_l <- vst(dds_adults, blind=TRUE)
pcaData1<-plotPCA(vstcounts_l, intgroup="condition",returnData=TRUE)
percentVar1 <- round(100 * attr(pcaData1, "percentVar"))

pcaData1$condition<-factor(pcaData1$condition,levels=c("hive_control","lab_food_nothing","adult_gut_transfer","bee_bread_addition","larva_gut_transfer","larva_gut_bee_bread"), labels=c("Hive","C","AG","BB","LG","LGBB"))
levels(pcaData1$condition)

ggplot(pcaData1, aes(PC1, PC2, color=condition)) +
geom_point(size=4)+xlab(paste0("PC1: ",percentVar[1],"% variance")) +
ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
coord_fixed()+theme_bw()+theme(legend.title = element_blank())+ scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"))+ theme(legend.text = element_text(face="bold", size=15))+theme(axis.text = element_text(size=15))+theme(axis.title = element_text(size=16))#+ylim(-50,50)+xlim(-75,75)
ggsave("Adult-output/pca_adults.png", height = 4, width = 4)
```

### MA plots treatments vs hive control

```{r MA_plots_adults, fig.width=4, fig.height=4}

#Shrinkage of effect size (LFC estimates) is useful for visualization and ranking of genes.

resultsNames(dds_adults)

resLFC_adults_AG <- lfcShrink(dds_adults,coef="condition_adult_gut_transfer_vs_hive_control", type="apeglm")
resLFC_adults_BB <- lfcShrink(dds_adults, coef="condition_bee_bread_addition_vs_hive_control", type="apeglm")
resLFC_adults_C <- lfcShrink(dds_adults, coef="condition_lab_food_nothing_vs_hive_control", type="apeglm")
resLFC_adults_LGBB <- lfcShrink(dds_adults, coef="condition_larva_gut_bee_bread_vs_hive_control", type="apeglm")
resLFC_adults_LG <- lfcShrink(dds_adults, coef="condition_larva_gut_transfer_vs_hive_control", type="apeglm")

#### make graph combined with one legend

library(ggpubr)

AG<-ggmaplot(resLFC_adults_AG, fdr = 0.05, fc = 1, size = 1.2,palette = c("#B31B21", "#1465AC", "darkgray"),genenames = as.vector(resLFC_adults_AG$name),top = 0,legend="top",label.select = NULL)+coord_cartesian(xlim = c(0, 18))+ scale_y_continuous(limits=c(-8, 8))+ theme(axis.text.x = element_text(size=16),axis.text.y = element_text(size=15),axis.title.x = element_text(size=17),axis.title.y = element_text(size=17),axis.line = element_line(size = 1, colour="gray20"),axis.ticks = element_line(size = 1, colour="gray20"))+guides(color = guide_legend(override.aes = list(size = c(3,3,3))))+theme(legend.position = c(0.70, 0.12),legend.text=element_text(size=14,face="bold"),legend.background = element_rect(fill="transparent"))+theme(plot.title = element_text(size=18, colour="gray30", face="bold",hjust=0.06, vjust=-5))+labs(title="Adult gut addition")

BB<-ggmaplot(resLFC_adults_BB, fdr = 0.05, fc = 1, size = 1.2,palette = c("#B31B21", "#1465AC", "darkgray"),genenames = as.vector(resLFC_adults_BB$name),top = 0)+coord_cartesian(xlim = c(0, 18))+ scale_y_continuous(limits=c(-8, 8))+ theme(axis.text.x = element_text(size=16),axis.text.y = element_text(size=15),axis.title.x = element_text(size=17),axis.title.y = element_text(size=17),axis.line = element_line(size = 1, colour="gray20"),axis.ticks = element_line(size = 1, colour="gray20"))+guides(color = guide_legend(override.aes = list(size = c(3,3,3))))+theme(legend.position = c(0.70, 0.12),legend.text=element_text(size=14,face = "bold"),legend.background = element_rect(fill="transparent"))+theme(axis.line.y = element_line(color = "white"))+ theme(axis.title.y = element_blank(),axis.text.y = element_blank(),axis.ticks.y = element_blank(),axis.title.x = element_blank(),axis.text.x = element_blank())+theme(plot.title = element_text(size=18, colour="gray30", face="bold",hjust=0.06, vjust=-5))+labs(title="Bee bread addition")

C<-ggmaplot(resLFC_adults_C, fdr = 0.05, fc = 1, size = 1.2,palette = c("#B31B21", "#1465AC", "darkgray"),genenames = as.vector(resLFC_adults_C$name),top = 0)+coord_cartesian(xlim = c(0, 18))+ scale_y_continuous(limits=c(-8, 8))+ theme(axis.text.x = element_text(size=16),axis.text.y = element_text(size=15),axis.title.x = element_text(size=17),axis.title.y = element_text(size=17),axis.line = element_line(size = 1, colour="gray20"),axis.ticks = element_line(size = 1, colour="gray20"))+guides(color = guide_legend(override.aes = list(size = c(3,3,3))))+theme(legend.position = c(0.70, 0.12),legend.text=element_text(size=14,face = "bold"),legend.background = element_rect(fill="transparent"))+theme(axis.line.y = element_line(color = "white"))+ theme(axis.title.y = element_blank(),axis.text.y = element_blank(),axis.ticks.y = element_blank(),axis.title.x = element_blank(),axis.text.x = element_blank())+theme(plot.title = element_text(size=18, colour="gray30", face="bold",hjust=0.06, vjust=-5))+labs(title="No addition")

LG<-ggmaplot(resLFC_adults_LG, fdr = 0.05, fc = 1, size = 1.2,palette = c("#B31B21", "#1465AC", "darkgray"),genenames = as.vector(resLFC_adults_LG$name),top = 0)+coord_cartesian(xlim = c(0, 18))+ scale_y_continuous(limits=c(-8, 8))+ theme(axis.text.x = element_text(size=16),axis.text.y = element_text(size=15),axis.title.x = element_text(size=17),axis.title.y = element_text(size=17),axis.line = element_line(size = 1, colour="gray20"),axis.ticks = element_line(size = 1, colour="gray20"))+guides(color = guide_legend(override.aes = list(size = c(3,3,3))))+theme(legend.position = c(0.70, 0.12),legend.text=element_text(size=14,face = "bold"),legend.background = element_rect(fill="transparent"))+theme(axis.line.y = element_line(color = "white"))+ theme(axis.title.y = element_blank(),axis.text.y = element_blank(),axis.ticks.y = element_blank(),axis.title.x = element_blank(),axis.text.x = element_blank())+theme(plot.title = element_text(size=18, colour="gray30", face="bold",hjust=0.06, vjust=-5))+labs(title="Larva gut addition")
  
LGBB<-ggmaplot(resLFC_adults_LGBB, fdr = 0.05, fc = 1, size = 1.2,palette = c("#B31B21", "#1465AC", "darkgray"),genenames = as.vector(resLFC_adults_LGBB$name),top = 0)+coord_cartesian(xlim = c(0, 18))+ scale_y_continuous(limits=c(-8, 8))+ theme(axis.text.x = element_text(size=16),axis.text.y = element_text(size=15),axis.title.x = element_text(size=17),axis.title.y = element_text(size=17),axis.line = element_line(size = 1, colour="gray20"),axis.ticks = element_line(size = 1, colour="gray20"))+guides(color = guide_legend(override.aes = list(size = c(3,3,3))))+theme(legend.position = c(0.70, 0.12),legend.text=element_text(size=14,face = "bold"),legend.background = element_rect(fill="transparent"))+theme(axis.line.y = element_line(color = "white"))+ theme(axis.title.y = element_blank(),axis.text.y = element_blank(),axis.ticks.y = element_blank(),axis.title.x = element_blank(),axis.text.x = element_blank())+theme(plot.title = element_text(size=18, colour="gray30", face="bold",hjust=0.06, vjust=-5))+labs(title="Larva gut & bee bread")

library(patchwork)
List <- list(AG,BB,C,LG,LGBB)
Plot <- wrap_plots(List,nrow=1)
ggsave("Adult-output/adults_MA_all.png", height = 5, width = 15)
```

### extract significant differently expressed genes

```{r extract sig genes adults}

functions <- read_csv("data_files/annotationsBiomart.csv", col_types = cols()) %>% dplyr::select(Gene_ID=ensembl_gene_id,description=description)

res_LG_sig <- results(dds_adults, tidy=TRUE, contrast=c("condition", "larva_gut_transfer", "hive_control"))
write.csv(as.data.frame(res_LG_sig), file="Adult-output/res_LG_all.csv")
res_LG_sig<- subset(res_LG_sig, padj < 0.05)
colnames(res_LG_sig)[which(names(res_LG_sig) == "row")] <- "Gene_ID"
res_LG_sig<-merge(res_LG_sig, functions, by="Gene_ID", type="left", match="first")
write.csv(as.data.frame(res_LG_sig), file="Adult-output/res_LG_sig.csv")

#How many adjusted p-values were less than 0.05?
sum(res_LG_sig$padj < 0.05, na.rm=TRUE)

res_AG_sig <- results(dds_adults, tidy=TRUE, contrast=c("condition", "adult_gut_transfer", "hive_control"))
write.csv(as.data.frame(res_AG_sig), file="Adult-output/res_AG_all.csv")
res_AG_sig<- subset(res_AG_sig, padj < 0.05)
colnames(res_AG_sig)[which(names(res_AG_sig) == "row")] <- "Gene_ID"
res_AG_sig<-merge(res_AG_sig, functions, by="Gene_ID", type="left", match="first")
write.csv(as.data.frame(res_AG_sig), file="Adult-output/res_AG_sig.csv")
sum(res_AG_sig$padj < 0.05, na.rm=TRUE)

res_C_sig <- results(dds_adults, tidy=TRUE, contrast=c("condition", "lab_food_nothing", "hive_control"))
write.csv(as.data.frame(res_C_sig), file="Adult-output/res_C_all.csv")
res_C_sig<- subset(res_C_sig, padj < 0.05)
colnames(res_C_sig)[which(names(res_C_sig) == "row")] <- "Gene_ID"
res_C_sig<-merge(res_C_sig, functions, by="Gene_ID", type="left", match="first")
write.csv(as.data.frame(res_C_sig), file="Adult-output/res_C_sig.csv")
sum(res_C_sig$padj < 0.05, na.rm=TRUE)

res_LGBB_sig <- results(dds_adults, tidy=TRUE, contrast=c("condition", "larva_gut_bee_bread", "hive_control"))
write.csv(as.data.frame(res_LGBB_sig), file="Adult-output/res_LGBB_all.csv")
res_LGBB_sig<- subset(res_LGBB_sig, padj < 0.05)
colnames(res_LGBB_sig)[which(names(res_LGBB_sig) == "row")] <- "Gene_ID"
res_LGBB_sig<-merge(res_LGBB_sig, functions, by="Gene_ID", type="left", match="first")
write.csv(as.data.frame(res_LGBB_sig), file="Adult-output/res_LGBB_sig.csv")
sum(res_LGBB_sig$padj < 0.05, na.rm=TRUE)

res_BB_sig <- results(dds_adults, tidy=TRUE, contrast=c("condition", "bee_bread_addition", "hive_control"))
write.csv(as.data.frame(res_BB_sig), file="Adult-output/res_BB_all.csv")
res_BB_sig<- subset(res_BB_sig, padj < 0.05)
colnames(res_BB_sig)[which(names(res_BB_sig) == "row")] <- "Gene_ID"
res_BB_sig<-merge(res_BB_sig, functions, by="Gene_ID", type="left", match="first")
write.csv(as.data.frame(res_BB_sig), file="Adult-output/res_BB_sig.csv")
sum(res_BB_sig$padj < 0.05, na.rm=TRUE)
```


#### Go annotation adults


```{r GO2}
library(AnnotationDbi)
library(GSEABase)
library(Category)

go <- read_csv("data_files/GO/GOannotationsBiomart_mod.csv", col_types = cols()) %>% mutate(evidence = "ISS") %>% dplyr::select(GO=go_id, evidence,gene=ensembl_gene_id)
universe <- unique(go$gene)
goFrame <- GOFrame(as.data.frame(go %>% dplyr::filter(gene %in% universe)), organism="Apis mellifera")
goAllFrame <- GOAllFrame(goFrame)
gsc <- GeneSetCollection(goAllFrame, setType = GOCollection())
```

```{r GO_AG3}
#up
AG_up <- read.csv("Adult-output/res_AG_sig_up.csv")

DsxModuleGo <- hyperGTest(GSEAGOHyperGParams(name = "enriched in AG treatment",
    geneSetCollection=gsc,geneIds = intersect(AG_up$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))

summary(DsxModuleGo) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 

GO_enrich.AG <- as.data.frame(summary(DsxModuleGo))
GO_enrich.AG %>% 
arrange(Pvalue) %>% 
write.csv(file = "Adult-output/Go/GO_term_enrichment_AG_up.csv")

### make list
hub_genesGO <- read.csv("Adult-output/Go/GO_term_enrichment_AG_up.csv")
# keep only highly significant terms
hub_genesGO <- hub_genesGO %>%
filter(Pvalue < 0.05) 

#plot it
g <- hub_genesGO %>%
  mutate(logPv = log(Pvalue))
ggplot(g, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("AG up compared to Hive")+theme(axis.title.y = element_blank())
ggsave("Adult-output/Go/AG_up.png", height = 8, width = 9)

## the one down-regulated gene not working for GOterm
```


```{r GO_BB}

#up
BB_up <- read.csv("Adult-output/res_BB_sig_up.csv")

DsxModuleGo3 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in BB treatment",
    geneSetCollection=gsc,geneIds = intersect(BB_up$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))

summary(DsxModuleGo3) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 

GO_enrich.BB <- as.data.frame(summary(DsxModuleGo3))
GO_enrich.BB %>% 
arrange(Pvalue) %>% 
write.csv(file = "Adult-output/Go/GO_term_enrichment_BB_up.csv")

hub_genesGO <- read.csv("Adult-output/Go/GO_term_enrichment_BB_up.csv")
hub_genesGO <- hub_genesGO %>%
filter(Pvalue < 0.05) 
g <- hub_genesGO %>%
  mutate(logPv = log(Pvalue))
ggplot(g, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("BB up compared to Hive")+theme(axis.title.y = element_blank())
ggsave("Adult-output/Go/BB_up.png", height = 8, width = 9)

#down
BB_down <- read.csv("Adult-output/res_BB_sig_down.csv")

DsxModuleGo2 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in BB treatment",
    geneSetCollection=gsc,geneIds = intersect(BB_down$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))
summary(DsxModuleGo2) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 
GO_enrich.BB2 <- as.data.frame(summary(DsxModuleGo2))
GO_enrich.BB2 %>% 
arrange(Pvalue) %>% 
write.csv(file = "Adult-output/Go/GO_term_enrichment_BB_down.csv")
hub_genesGO2 <- read.csv("Adult-output/Go/GO_term_enrichment_BB_down.csv")
hub_genesGO2 <- hub_genesGO2 %>%
filter(Pvalue < 0.05) 

g2 <- hub_genesGO2 %>%
  mutate(logPv = log(Pvalue))
ggplot(g2, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("BB down compared to Hive")+theme(axis.title.y = element_blank())
ggsave("Adult-output/Go/BB_down.png", height = 8, width = 9)
```

```{r GO_C}

#up
#not working for GOterm

#down
C_down <- read.csv("Adult-output/res_C_sig_down.csv")

DsxModuleGo5 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in C treatment",
    geneSetCollection=gsc,geneIds = intersect(C_down$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))
summary(DsxModuleGo5) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 
GO_enrich.C2 <- as.data.frame(summary(DsxModuleGo5))
GO_enrich.C2 %>% 
arrange(Pvalue) %>% 
write.csv(file = "Adult-output/Go/GO_term_enrichment_C_down.csv")
```


```{r GO_LG}

#up
LG_up <- read.csv("Adult-output/res_LG_sig_up.csv")

DsxModuleGo6 <- hyperGTest(GSEAGOHyperGParams(name = "enriched in LG treatment",
    geneSetCollection=gsc,geneIds = intersect(LG_up$Gene_ID,universe), universeGeneIds=universe, ontology = "BP",pvalueCutoff = 0.05,conditional = FALSE,testDirection = "over"))

summary(DsxModuleGo6) %>% kable("html", digits = 3) %>% kable_styling() %>%  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width = T) %>% scroll_box(height = "400px") 

GO_enrich.LG <- as.data.frame(summary(DsxModuleGo6))
GO_enrich.LG %>% 
arrange(Pvalue) %>% 
write.csv(file = "Adult-output/Go/GO_term_enrichment_LG_up.csv")

hub_genesGO <- read.csv("Adult-output/Go/GO_term_enrichment_LG_up.csv")
hub_genesGO <- hub_genesGO %>%
filter(Pvalue < 0.05) 
g <- hub_genesGO %>%
  mutate(logPv = log(Pvalue))
ggplot(g, aes(x = 0, y = Term, size = Count, color = logPv)) +
  geom_point(alpha = 0.9) +
  theme_classic()+ggtitle("LG up compared to Hive")+theme(axis.title.y = element_blank())
ggsave("Adult-output/Go/LG_up.png", height = 8, width = 9)

#down
#not doable with GO term analysis
```

```{r GO_LGBB}
#up
#down
#not working for GO term analysis with p-value filtering of 0.05
```
