---
title: "R analysis larva-adult"
author: "Vienna"
date: "4/27/2021"
output: html_document
  pdf_document: default
  keep_md: true
  html_document: default
self_contained: no
thumbnails: no
keep_md: yes
---

### Background
#### The Nov 2018 conducted experiment at OIST tested for a connection of the host-associated microbiome between the two, by metamorphosis seperated life stages of honey bees; larval stage and adult stage. Therefore, the experiment is composed of two parts: the larvae rearing part and the adult part. 

#### In short, to create larvae with significantly different microbiomes, larvae were picked from a hive at age <24 hours. In the lab they were fed with different diets to transfer microbiome. The main treatments are: <span style="color:red">**C = no_control**</span> -> the bees did just get the lab-larvae food, <span style="color:red">**LG = Larva gut**</span> -> pooled gut content of 5 last stage larvae from the same hive transferred on two days (day 2 and day 3), <span style="color:red">**AG = Adult gut**</span> -> same procedure as for LG just gut content of nurse bees used, <span style="color:red">**BB = Bee bread**</span> -> freshly collected bee bread mixed with larva food in field-realistic concentration (on day 3, 4, 5 (AFTER fist sampling day three as larvae showed high mortality in trials before when feeding a day earlier)), <span style="color:red">**LGBB = Larva gut and bee bread**</span> -> both, larvae gut and bee bread transferred at the respective LG and BB time points, <span style="color:red">**Hive = hive control**</span> -> larvae from the same frame raised under natural conditions in the hive.
#### Sampling of all larvae: day zero (start point) = three single larvae, three pools of 5 larvae each, day three (five samples per treatment), day four (three samples per treatment), day six (~five samples per treatment (six for LG and LGBB and four for AG))


#### Emerged adults from all treatments were distributed to three cages and bees got on two days a mix of each 10 nurse guts from the same hive (pool of this mix also sequenced as one sample = NG_mix). After one week the adults were sampled, surface sterilized and the guts were dissected for later DNA extraction. Extracted were most of the time 3 samples per cage (in two cases only 2 as bees died or sample extraction did not work)

### Data types: - experimental survival data, - 16S amplicon sequence data output from Qiime2, - qPCR data, - RNA sequence count output from kallisto


```{r setup, include = FALSE}
library(RSQLite)
library(DECIPHER)
library(DESeq2)
library(gridExtra)
library(ggnewscale) ##new_scale()
library(ggpubr) # ggarrange and ggerrorplot function
library(ggthemes)
library(grid) # rasterGrob()
library(vegan)
library(phyloseq)
library(png)
library(qgraph) # graph layout
library(ggsignif)
library(btools)
library(RColorBrewer)
library(tidyverse) # includes: ggplot2, dplyr, tidyr, readr, purr, tibble, stringr, forcats
library(knitr)
knitr::opts_chunk$set(warning=FALSE, message=FALSE)
set.seed(42)
do_tech_analysis <- TRUE
```

# 16S microbiome analysis


## 16S Adults samples analysis

```{r 16S_load}
Svtab <- read.delim("microbiome/adults/data_files/feature_table.txt")
# replace . in colnames with - to match metadata.txt
colnames(Svtab)  <- gsub("\\.", "-", colnames(Svtab))
row.names(Svtab)<-Svtab[[1]] #make OTU ID the row names
Svtab<-Svtab[,-(1)] # remove OTU ID column
fmat <- as.matrix(Svtab)
OTU = otu_table(fmat, taxa_are_rows = TRUE)

#read in taxonomy
taxonomy=read.csv("microbiome/adults/data_files/taxonomy.csv",sep=",",row.names=1)
taxonomy <- cbind(taxonomy, ASV = paste0("ASV", sprintf("%04d", 1:nrow(taxonomy))))
taxonomy=as.matrix(taxonomy)
TAX=tax_table(taxonomy)

metatable <- read.delim("microbiome/adults/data_files/sample-metadata.txt") 
#View(metatable)
row.names(metatable) <- metatable[[1]]
metatable<- metatable[,(-1)]
META <- sample_data(metatable)

ps_adults1<- phyloseq(OTU, TAX, META)

# change taxonomy header 
colnames(tax_table(ps_adults1)) <- c(D0 = "Kingdom", D1 = "Phylum", D2 = "Class", 
                              D3 = "Order", D4 = "Family", D5 = "Genus", D6 = "Species", ASV = "ASV")

#The total number of ASVs in the whole dataset is 
length(taxa_names(ps_adults1))

#get rid of things we do not want
ps_adults1 = subset_taxa(ps_adults1, Kingdom =="Bacteria")
ps_adults1 <- prune_taxa(taxa_sums(ps_adults1) > 0, ps_adults1)
ps_adults1<-subset_taxa(ps_adults1, (Order!="Chloroplast"))
ps_adults1<-subset_taxa(ps_adults1, (Family!="Mitochondria"))
length(taxa_names(ps_adults1))
#reduces total taxa numbers from 415 to 396 (minus 19)

# mean, max and min of sample read counts
smin <- min(sample_sums(ps_adults1))
smean <- mean(sample_sums(ps_adults1))
smax <- max(sample_sums(ps_adults1))

# printing the results
cat("The minimum sample read count is:",smin)
cat("The average sample read count is:",smean)
cat("The maximum sample read count is:",smax)

ps_adults = subset_samples(ps_adults1, treatment != "inoculum")
```

### Plot rarefaction curve
```{r rarefaction_curve}
rare<- ps_adults1
smin <- min(sample_sums(rare))
cat("The minimum sample read count is:",smin)

# rarefy
set.seed(42)
ps.rarefied = rarefy_even_depth(rare, rngseed=1, sample.size=42000, replace=F)
library(ranacapa)
cbPalette <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7","maroon","khaki4")

ggrare(ps.rarefied, step = 100, se = FALSE, color="treatment")+ scale_colour_manual(values=cbPalette)+geom_line(size=1)+ theme_bw()+theme(axis.line = element_line(colour = "black"))+ theme(legend.title = element_blank(),
                legend.background = element_blank(),
                legend.key = element_blank()) + theme(text = element_text(size = 22))+theme(strip.text = element_text(face="bold", size=20))+ theme(axis.title.x=element_blank())+ggtitle("Rarefaction adult samples")+theme(plot.title = element_text(hjust = 0.5, size=20, face="italic"))+theme(legend.text=element_text(size=17, face="italic"))

ggsave("microbiome/adults/figures/rarefaction.png", height = 5, width = 12)
```

*** Supplementary figure: Rarefaction curves were used as a qualitative method to estimate the species richness as a function of sequencing depth for all samples. Rarefaction curves reached their asymptotes for all samples, suggesting that saturation in sequencing was achieved.


##  Alpha diversity 

```{r alpha_Shannon_adults, fig.heigt=13, fig.width=8}
alpha<- ps_adults
alpha1_ra <- rarefy_even_depth(alpha,sample.size=42000, replace=FALSE, rngseed = 1) 
alpha1_ra<-prune_taxa(taxa_sums(alpha1_ra) > 0, alpha1_ra)

sample_data(alpha1_ra)$treatment<-factor(sample_data(alpha1_ra)$treatment,levels=c("Hive","C","AG","LG","BB","LGBB"))
                                         #, labels=c("Hive control","RJ","RJ \n +adult gut", "RJ \n +larvae gut", "RJ \n +bee bread","RJ \n +larvae gut \n +bee bread"))
levels(sample_data(alpha1_ra)$treatment)

sigFunc = function(x){
  if(x < 0.001){"***"} 
  else if(x < 0.01){"**"}
  else if(x < 0.05){"*"}
  else{NA}}

alpha_meas = c("Shannon")
p2 <- plot_richness(alpha1_ra,"treatment",color="treatment", measures=alpha_meas)
p2$layers <- p2$layers[-1]
p2<-p2 + geom_boxplot(data=p2$data, aes(x=treatment, y=value),show_guide=FALSE, alpha=0.1)+geom_point(position = position_jitter(w = 0.2, h = 0.1),show_guide=FALSE) +
  geom_signif(comparisons=list(c("Hive", "C"), c("Hive", "AG"),c("Hive", "LG"),c("Hive", "BB"),c("Hive", "LGBB")),tip_length = 0.02,vjust=0.5, color=("grey20"),map_signif_level=sigFunc, y_position = c(3.9,3.9,3.9,3.6,3.9), margin_top = 0.1, textsize=6, test = wilcox.test)
p2 <- p2 + guides(fill=guide_legend(title=element_blank()))+ylim(1.7,4)
p2 <- p2 + theme(plot.title = element_text(hjust = 0.5))+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"))
p2 <- p2+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+ theme(panel.background = element_blank())
p2 <- p2+theme(axis.line = element_line(colour = "black"))
p2<- p2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank()) +ylab("Shannon alpha diversity")
p2 <- p2+ theme(panel.border = element_blank())+theme(strip.text = element_blank())
p2<-p2 + theme(text = element_text(size = 16))+theme(strip.text = element_text(face="bold", size=15, color="gray40"))+ theme(axis.title.x=element_blank())+theme(axis.text.x = element_text(size=17,angle=-45))+theme(strip.background =element_rect(fill="white", color="white"))
ggsave("microbiome/adults/figures/alpha_Shannon_adults.png", height = 3.5, width = 6.5)
```
***The Shannon index  captures  information  about  both  the  species  richness  (number  of species) and the relative abundances of the species. Increased Shannon index means increased species diversity.


```{r alpha_observed_adults}
alpha<- ps_adults
alpha1_ra <- rarefy_even_depth(alpha,sample.size=42000, replace=FALSE, rngseed = 1) 
alpha1_ra<-prune_taxa(taxa_sums(alpha1_ra) > 0, alpha1_ra)

sample_data(alpha1_ra)$treatment<-factor(sample_data(alpha1_ra)$treatment,levels=c("Hive","C","AG","LG","BB","LGBB"))
levels(sample_data(alpha1_ra)$treatment)

alpha_meas = c("Observed")
p2 <- plot_richness(alpha1_ra,"treatment",color="treatment", measures=alpha_meas)
p2$layers <- p2$layers[-1]
p2<-p2 + geom_boxplot(data=p2$data, aes(x=treatment, y=value),show_guide=FALSE, alpha=0.1)+geom_point(position = position_jitter(w = 0.2, h = 0.1),show_guide=FALSE)
#p2 <- p2 + guides(fill=guide_legend(title=element_blank()))+ylim(1.7,4)
p2 <- p2 + theme(plot.title = element_blank())+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"))
p2 <- p2+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+ theme(panel.background = element_blank())
p2 <- p2+theme(axis.line = element_line(colour = "black"))
p2<- p2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank()) +ylab("Observed species")
p2 <- p2+ theme(panel.border = element_blank())+ theme(axis.ticks.x = element_blank())
Observed_adults<-p2 + theme(text = element_text(size = 17))+theme(strip.text = element_blank())+ theme(axis.title.x=element_blank())+theme(strip.background =element_rect(fill="white", color="white"))+theme(axis.text.x = element_blank())+ theme(axis.title.y=element_blank())
ggsave("microbiome/adults/figures/alpha_Observed_adults.png", height = 3.5, width = 6.5)
```

```{r warnings=FALSE}
alpha<- ps_adults
alpha1_ra <- rarefy_even_depth(alpha,sample.size=42000, replace=FALSE, rngseed = 1) 

#Shannon

results = estimate_richness(alpha1_ra, measures = 'Shannon')
d = sample_data(alpha1_ra)

# calculate wilcox-test
H = results[d[,'treatment'] == 'Hive',]
C = results[d[,'treatment'] == 'C',]
AG = results[d[,'treatment'] == 'AG',]
LG = results[d[,'treatment'] == 'LG',]
BB = results[d[,'treatment'] == 'BB',]
LGBB = results[d[,'treatment'] == 'LGBB',]

wilcox.test(H, C)
wilcox.test(H, AG)
wilcox.test(H, LG)
wilcox.test(H, BB)
wilcox.test(H, LGBB)

pvalues<-c(0.18,0.15,0.45,0.008,0.78)
p.adjust(pvalues,method="fdr")

# observed

results1 = estimate_richness(alpha1_ra, measures = 'Observed')
d1 = sample_data(alpha1_ra)

# calculate wilcox-test
H1 = results1[d1[,'treatment'] == 'Hive',]
C1 = results1[d1[,'treatment'] == 'C',]
AG1 = results1[d1[,'treatment'] == 'AG',]
LG1 = results1[d1[,'treatment'] == 'LG',]
BB1 = results1[d1[,'treatment'] == 'BB',]
LGBB1= results1[d1[,'treatment'] == 'LGBB',]

wilcox.test(H1, C1)
wilcox.test(H1, AG1)
wilcox.test(H1, LG1)
wilcox.test(H1, BB1)
wilcox.test(H1, LGBB1)

pvalues<-c(0.59,0.03,0.27,0.11,0.77)
p.adjust(pvalues,method="fdr")

#statistics
rich = estimate_richness(alpha1_ra)
pairwise.wilcox.test(rich$Observed,sample_data(alpha1_ra)$treatment,p.adjust.method ="fdr")

pairwise.wilcox.test(rich$Shannon,sample_data(alpha1_ra)$treatment,p.adjust.method ="fdr")
```

# Beta diversity

### ordination plots
```{r NMDS_adults}
#NMDS on Bray-Curtis
ord2<- ps_adults
ord2 <- transform_sample_counts(ord2, function(OTU) {OTU / sum(OTU)})
sample_data(ord2)$treatment<-factor(sample_data(ord2)$treatment,levels=c("Hive","C","AG","LG","BB","LGBB"))
levels(sample_data(ord2)$treatment)
set.seed(1)
dist<-phyloseq::distance(ord2, method="bray") 
ordination = ordinate(ord2, method="NMDS", distance=dist)
cat("stress is:", ordination$stress)
p2 <- plot_ordination(ord2, ordination, color="treatment") +geom_point(size=2.5)
p2 <- p2 + guides(fill=guide_legend(title=element_blank()))
p2 <- p2 + theme(plot.title = element_text(hjust = 0.5))+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"),guide = guide_legend(override.aes = list(shape = c(rep(15, 6)),size=7,alpha=0.5)))
p2 <- p2+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+ theme(panel.background = element_blank())
p2 <- p2+theme(axis.line = element_line(colour = "black"))
p2<- p2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank())
p2 <- p2+ theme(panel.border = element_blank())
p2<-p2 + theme(text = element_text(size = 16.5))+theme(strip.text = element_blank())
p2<-p2+theme(legend.position="bottom")+theme(strip.background =element_rect(fill="white", color="white"))
p2<-p2+stat_ellipse(aes(group = treatment),size=1.5, alpha=0.5)+ theme(panel.spacing.x=unit(5.5, "lines"))+coord_cartesian(xlim = c(-0.4, 1))+coord_cartesian(ylim = c(-0.8, 0.8))+ theme(axis.title.y = element_text(margin = margin(t = 0, r = 0, b = 0, l = 20)))
ggsave("microbiome/adults/figures/NMDS_adults.png", height = 5.5, width = 8)
```

```{r PCoA_adults}
#PCoA on Bray-Curtis
ord2<- ps_adults
ord2 = subset_samples(ord2, treatment != "AG_Larva" & treatment != "inoculum" & treatment != "old")
ord2 <- transform_sample_counts(ord2, function(OTU) {OTU / sum(OTU)})
sample_data(ord2)$treatment<-factor(sample_data(ord2)$treatment,levels=c("Hive","C","AG","LG","BB","LGBB"),labels=c("Hive control", "+no addition (C)", "+adult gut (AG)", "+larvae gut (LG)", "+bee bread (BB)", "+larvae gut & bee bread (LGBB)"))
levels(sample_data(ord2)$treatment)
set.seed(1)
dist<-phyloseq::distance(ord2, method="bray") 
ordination = ordinate(ord2, method="PCoA", distance=dist)
p2 <- plot_ordination(ord2, ordination, color="treatment") +geom_point(size=2.5)
p2 <- p2 + guides(fill=guide_legend(title=element_blank()))
p2 <- p2 + theme(plot.title = element_text(hjust = 0.5))+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"),guide = guide_legend(override.aes = list(shape = c(rep(15, 6)),size=9,alpha=0.3)))
p2 <- p2+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+ theme(panel.background = element_blank())
p2 <- p2+theme(axis.line = element_line(colour = "black"))
p2<- p2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank())
p2 <- p2+ theme(panel.border = element_blank())
p2<-p2 + theme(text = element_text(size = 16))+theme(strip.text = element_blank())
p2<-p2+theme(legend.position="bottom")+theme(strip.background =element_rect(fill="white", color="white"))
PCoA_adults<-p2+stat_ellipse(aes(group = treatment),size=1.5, alpha=0.5)+ theme(panel.spacing.x=unit(5.5, "lines"))+ theme(axis.title.y = element_text(margin = margin(t = 0, r = 0, b = 0, l = 20)))+ theme(legend.text = element_text(face="bold", size=17,color="gray9"))
ggsave("microbiome/adults/figures/PCoA_adults.png", height = 5.5, width = 8)
```
## Beta diversity stats

#### test difference between treatments and hive control
```{r}
compare=ps_adults
test2 <- transform_sample_counts(compare, function(OTU) {OTU / sum(OTU)})
test2 <- prune_taxa(taxa_sums(test2) > 0, test2)

df = as(sample_data(test2), "data.frame")
set.seed(53)
d = phyloseq::distance(test2, "bray")
adonis = adonis(d ~ treatment, df, perm=999)
adonis

#testing interaction between treatment and cage
set.seed(53)
d = phyloseq::distance(test2, "bray")
adonis = adonis(d ~ treatment*cage, df,perm=999)
adonis

subs <- subset_samples(test2, treatment %in% c("Hive", "C"))
subs <- prune_taxa(taxa_sums(subs) > 0, subs)
metadata <- as(sample_data(subs), "data.frame")
adonis(distance(subs, method="bray") ~ treatment,data = metadata, perm=999)

subs1 <- subset_samples(test2, treatment %in% c("Hive", "AG"))
subs1 <- prune_taxa(taxa_sums(subs1) > 0, subs1)
metadata <- as(sample_data(subs1), "data.frame")
adonis(distance(subs1, method="bray") ~ treatment,data = metadata, perm=999)

subs2 <- subset_samples(test2, treatment %in% c("Hive", "LG"))
subs2 <- prune_taxa(taxa_sums(subs2) > 0, subs2)
metadata <- as(sample_data(subs2), "data.frame")
adonis(distance(subs2, method="bray") ~ treatment,data = metadata, perm=999)

subs <- subset_samples(test2, treatment %in% c("Hive", "BB"))
subs <- prune_taxa(taxa_sums(subs) > 0, subs)
metadata <- as(sample_data(subs), "data.frame")
adonis(distance(subs, method="bray") ~ treatment,data = metadata, perm=999)

subs3 <- subset_samples(test2, treatment %in% c("Hive", "LGBB"))
subs3 <- prune_taxa(taxa_sums(subs3) > 0, subs3)
metadata <- as(sample_data(subs3), "data.frame")
adonis(distance(subs3, method="bray") ~ treatment,data = metadata, perm=999)

pvalues<-c(0.41,0.29,0.47,0.1,0.61)
p.adjust(pvalues,method="fdr")
```

# Taxonomy plots and stats

```{r set_up_for_taxa_plots}
tax <-ps_adults
tax <- prune_taxa(taxa_sums(tax) > 0, tax)

ps_bdiv2 <- transform_sample_counts(tax, function(OTU) {OTU / sum(OTU)})
tax_all2<- ps_bdiv2
ps_pd2 <- tax_glom(tax_all2, taxrank = 'Genus')

low1pc_reads <- max(taxa_sums(ps_pd2)) * 1 / 100
low1pc_indices <- which(taxa_sums(ps_pd2) < low1pc_reads)
length(low1pc_indices)
taxa_names(ps_pd2)[low1pc_indices[1]] <- "other"
taxa3 <- merge_taxa(ps_pd2, low1pc_indices, "other")
tax_table(taxa3)["other", ] <- c("other")

#build a custom color palette for phyloseq object
getPalette = colorRampPalette(brewer.pal(9, "Set3"))
speciesList = unique(tax_table(taxa3)[,"Genus"])
speciesPalette = getPalette(length(speciesList))
names(speciesPalette) = speciesList

ps_df <- psmelt(taxa3)
ps_df_sum <- ps_df %>%
  group_by(treatment,Genus) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(treatment) %>%
  mutate(relative_abundance = Abundance / sum(Abundance)) %>%
  ungroup() %>%
  mutate(Genus = fct_reorder(Genus, Abundance, .fun = sum, .desc = TRUE))
levels(ps_df_sum$Genus)

neworder <- c("Hive","C","AG","LG","BB","LGBB")
change <- c("Hive\n  ","C\n  ","AG\n  ","LG\n  ","BB\n  ","LGBB\n  ")
ps_df_sum2 <- arrange(transform(ps_df_sum,treatment=factor(treatment,levels=neworder,labels=change)),treatment)

p<-ggplot(data = ps_df_sum2, aes(x = treatment, y = relative_abundance, fill = Genus)) +
  geom_bar(stat = "identity")+ scale_fill_manual(values= speciesPalette)+ scale_y_continuous(expand = c(0,0))+ylab("relative abundance")
#p<-p+ theme(legend.position="bottom")
p<-p+ theme(legend.text=element_text(size=10.5, face = "italic"))+theme(legend.key = element_rect(color = NA, fill = NA),legend.key.size = unit(0.5, "cm"))+theme(legend.title = element_blank())+theme(legend.position="bottom")
taxa2<-p+theme(axis.title.y = element_text(size=12, face="bold"))+theme(axis.text.y = element_text(size=11, face="bold"))+theme(axis.text.x = element_text(size=11, face="bold",vjust=0.5))+theme(axis.title.x = element_blank())+theme(strip.background =element_rect(fill="white", color="white"))#+theme(axis.text.x =element_blank())
taxa2+guides(fill=guide_legend(nrow=3))
ggsave("microbiome/adults/figures/taxa_adults.png", height = 3.8, width = 4.7)
```


```{r individual}
tax <-ps_adults
tax <- prune_taxa(taxa_sums(tax) > 0, tax)

ps_bdiv2 <- transform_sample_counts(tax, function(OTU) {OTU / sum(OTU)})
tax_all2<- ps_bdiv2
ps_pd2 <- tax_glom(tax_all2, taxrank = 'Genus')
low1pc_reads <- max(taxa_sums(ps_pd2)) * 1 / 100
low1pc_indices <- which(taxa_sums(ps_pd2) < low1pc_reads)
length(low1pc_indices)
taxa_names(ps_pd2)[low1pc_indices[1]] <- "other"
taxa3 <- merge_taxa(ps_pd2, low1pc_indices, "other")
tax_table(taxa3)["other", ] <- c("other")

#build a custom color palette for phyloseq object
getPalette = colorRampPalette(brewer.pal(9, "Set3"))
speciesList = unique(tax_table(taxa3)[,"Genus"])
speciesPalette = getPalette(length(speciesList))
names(speciesPalette) = speciesList

ps_df <- psmelt(taxa3)
ps_df_sum <- ps_df %>%
  group_by(Sample,treatment,Genus) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(Sample,treatment) %>%
  mutate(relative_abundance = Abundance / sum(Abundance)) %>%
  ungroup() %>%
  mutate(Genus = fct_reorder(Genus, Abundance, .fun = sum, .desc = TRUE))
levels(ps_df_sum$Genus)

neworder <- c("Hive","C","AG","LG","BB","LGBB")
change <- c("Hive\n  ","C\n  ","AG\n  ","LG\n  ","BB\n  ","LGBB\n  ")
ps_df_sum2 <- arrange(transform(ps_df_sum,treatment=factor(treatment,levels=neworder,labels=change)),treatment)

p<-ggplot(data = ps_df_sum2, aes(x = Sample, y = relative_abundance, fill = Genus)) +
  geom_bar(stat = "identity")+ scale_fill_manual(values= speciesPalette)+ scale_y_continuous(expand = c(0,0))+ylab("relative abundance")
p<-p+ theme(legend.text=element_text(size=10.5, face = "italic"))+theme(legend.key = element_rect(color = NA, fill = NA),legend.key.size = unit(0.5, "cm"))+theme(legend.title = element_blank())+theme(legend.position="bottom")
taxa2<-p+theme(axis.title.y = element_text(size=12, face="bold"))+theme(axis.text.y = element_text(size=11, face="bold"))+theme(axis.text.x = element_text(size=7, face="bold",vjust=0.5,angle = 90))+theme(axis.title.x = element_blank())+theme(strip.background =element_rect(fill="white", color="white"))+theme(strip.text = element_text(size=10, face="bold",vjust = -5))
taxa2<-taxa2+guides(fill=guide_legend(nrow=3))
taxa2<-taxa2+facet_grid(~treatment, scale='free_x', space = "free_x")
#ggsave("adults/figures/taxa_adults_individuals.png", height = 4, width = 9)

taxa2<-taxa2+theme(axis.text.x = element_blank(), axis.ticks.x=element_blank())
#taxa2+facet_grid(~treatment, scale='free_x', space = "free_x")
ggsave("microbiome/adults/figures/taxa_adults_individuals.png", height = 3.5, width = 7)
```

```{r inoculum}
tax <-ps_adults1
tax = subset_samples(tax, treatment == "inoculum")

ps_bdiv2 <- transform_sample_counts(tax, function(OTU) {OTU / sum(OTU)})
tax_all2<- ps_bdiv2
ps_pd2 <- tax_glom(tax_all2, taxrank = 'Genus')

low1pc_reads <- max(taxa_sums(ps_pd2)) * 1 / 100
low1pc_indices <- which(taxa_sums(ps_pd2) < low1pc_reads)
length(low1pc_indices)
taxa_names(ps_pd2)[low1pc_indices[1]] <- "other"
taxa3 <- merge_taxa(ps_pd2, low1pc_indices, "other")
tax_table(taxa3)["other", ] <- c("other")

#build a custom color palette for phyloseq object
getPalette = colorRampPalette(brewer.pal(9, "Set3"))
speciesList = unique(tax_table(taxa3)[,"Genus"])
speciesPalette = getPalette(length(speciesList))
names(speciesPalette) = speciesList

ps_df <- psmelt(taxa3)
ps_df_sum <- ps_df %>%
  group_by(treatment,Genus) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(treatment) %>%
  mutate(relative_abundance = Abundance / sum(Abundance)) %>%
  ungroup() %>%
  mutate(Genus = fct_reorder(Genus, Abundance, .fun = sum, .desc = TRUE))
levels(ps_df_sum$Genus)

p<-ggplot(data = ps_df_sum, aes(x = treatment, y = relative_abundance, fill = Genus)) +
  geom_bar(stat = "identity")+ scale_fill_manual(values= speciesPalette)+ scale_y_continuous(expand = c(0,0))+ylab("relative abundance")
#p<-p+ theme(legend.position="bottom")
p<-p+ theme(legend.text=element_text(size=10.5, face = "italic"))+theme(legend.key = element_rect(color = NA, fill = NA),legend.key.size = unit(0.5, "cm"))+theme(legend.title = element_blank())
taxa2<-p+theme(axis.title.y = element_text(size=12, face="bold"))+theme(axis.text.y = element_text(size=11, face="bold"))+theme(axis.text.x = element_text(size=11, face="bold",vjust=0.5))+theme(axis.title.x = element_blank())+theme(strip.background =element_rect(fill="white", color="white"))#+theme(axis.text.x =element_blank())
#taxa2+guides(fill=guide_legend(nrow=3))  
ggsave("microbiome/adults/figures/taxa_inoculum.png", height = 3.3, width = 3.4)
```

### abundance all main taxa

### Core taxa abundance

```{r total abundance}
ps2 <-ps_adults
ps2 <- rarefy_even_depth(ps2,sample.size=42000, replace=FALSE, rngseed = 1) 
ps3 <- prune_taxa(taxa_sums(ps2) > 0, ps2)
ps3 <- tax_glom(ps3, taxrank = 'Genus')
psOrd3 = subset_taxa(ps3, Genus=="Lactobacillus" | Genus=="Snodgrassella" | Genus=="Gilliamella" | Genus=="Bifidobacterium" | Genus=="Bombella" | Genus=="Commensalibacter" | Genus=="Frischella"| Genus=="Fructobacillus")

#Melt and plot
melt<-psmelt(psOrd3)
neworder3 <- c("Hive","C","AG","LG","BB","LGBB")
melt4 <- arrange(transform(melt, treatment=factor(treatment,levels=neworder3)),treatment)

a_mean <- melt4 %>% 
  group_by(treatment,Genus) %>% 
  summarize(mean_val = mean(Abundance))
a_mean2<-subset(a_mean, treatment =="Hive")

sigFunc = function(x){
  if(x < 0.001){"***"} 
  else if(x < 0.01){"**"}
  else if(x < 0.05){"*"}
  else{NA}}

p<-ggplot(data = melt4, aes(x = treatment, y = Abundance)) +
  geom_boxplot(aes(fill=Genus),alpha=0.5,lwd=0.7, position = position_dodge(width = 0.3), width=0.45,outlier.shape = NA) + geom_jitter(aes(color = Genus), height = 0, width = .1)+geom_hline(data= a_mean2, aes(yintercept=mean_val), width=5,size=1,linetype="dashed")+
  labs(x = "", y = "Abundance\n")+
  facet_grid(Genus~., scales = "free")+theme_bw()+
  geom_signif(comparisons=list(c("Hive", "C"), c("Hive", "AG"),c("Hive", "LG"),c("Hive", "BB"),c("Hive", "LGBB")),tip_length = 0.03,vjust=1.3, color="grey20", size=1.1,map_signif_level=sigFunc, textsize=9, test = wilcox.test,step_increase = 0.2)
p<-p+ theme(legend.position="right")+ylab("Total abundance")
p<-p+ theme(legend.text=element_text(size=29, face = "italic"))+theme(legend.key = element_rect(color = NA, fill = NA),legend.key.size = unit(0.9, "cm"))+theme(legend.title = element_blank())
abu<-p +theme(strip.text.x = element_text(size=25,face="bold",color = "grey 35"))+theme(strip.background =element_rect(fill="white", color="white"))
taxa_adults<-abu+theme(axis.title.y = element_text(size=25, face="bold"))+theme(axis.text.y = element_text(size=20, face="bold"))+theme(axis.text.x = element_text(size=25, angle=90,face="bold"))+theme(axis.title.x = element_blank())+ theme(strip.text.y = element_blank())
ggsave("microbiome/adults/figures/core_total_abundance.png", height = 18, width = 16)
``` 

```{r, dabest_abundance_init}
#devtools::install_github("mikheyev/dabestr")
library(dabestr)
microbeColors <- function (n) {
  mcols <- c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum")
  return(mcols[4 %/% n])
}
```

#### Lactobacillus as example -> same for Bifidobacterium, Frischella, Snodgrassella, Gilliamella, Bombella, Commensalibacter
```{r, dabest_abundance_Lactobacillus}
ps3 <-ps_adults
set.seed(42)
ps3 <- rarefy_even_depth(ps3,sample.size=42000, replace=FALSE, rngseed = 1) 

sample_data(ps3)$treatment<-factor(sample_data(ps2.2)$treatment,levels=c("Hive","C","AG","LG","BB","LGBB"))
levels(sample_data(ps3)$treatment)
ps3 <- tax_glom(ps3, taxrank = 'Genus')
psord = subset_taxa(ps3, Genus=="Lactobacillus") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabestr::dabest(melt, treatment,Abundance,
                             idx = c("Hive","C","AG","LG","BB","LGBB"),
                             paired = FALSE)
unpaired_mean_diff
mean_diff(unpaired_mean_diff)
plot(mean_diff(unpaired_mean_diff),rawplot.ylabel ="Abundance",tick.fontsize=16,axes.title.fontsize=18,effsize.ylabel = "Mean diff.", rawplot.markersize = 2)#, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum") <- customizing colors worked before, not anymore
```

### test spread of variance of groups
```{r betadisp1}
set.seed(53)
test<-ps_adults
test2 <- transform_sample_counts(test, function(OTU) {OTU / sum(OTU)})
cycle3 <- prune_taxa(taxa_sums(test2) > 0, test2)

#testing treatment
set.seed(53)
d_cycle3 = phyloseq::distance(cycle3, "bray")
df_cycle3 = as(sample_data(cycle3), "data.frame")
df_cycle3$treatment <- factor(df_cycle3$treatment , levels=c("Hive","C","AG","LG","BB","LGBB"))
groups <- df_cycle3[["treatment"]]
beta <- betadisper(d_cycle3, df_cycle3$treatment)
permutest(beta,pairwise = TRUE, permutations = 999)
anova(betadisper(d_cycle3, groups))
mod.HSD <- TukeyHSD(beta,conf.level = 0.95)
mod.HSD
plot(betadisper(d_cycle3, groups),main="Adults",lwd=2, label = TRUE,label.cex = 0.7,col=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"))
boxplot(betadisper(d_cycle3, groups),main="Adults",border=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"),xlab="")
```


# 16S Larvae


```{r load}
Svtab <- read.delim("microbiome/larvae/data_files/feature_table.txt")
# replace . in colnames with - to match metadata.txt
colnames(Svtab)  <- gsub("\\.", "-", colnames(Svtab))
row.names(Svtab)<-Svtab[[1]] #make OTU ID the row names
Svtab<-Svtab[,-(1)] # remove OTU ID column
fmat <- as.matrix(Svtab)
OTU = otu_table(fmat, taxa_are_rows = TRUE)

#read in taxonomy
taxonomy=read.csv("microbiome/larvae/data_files/taxonomy.csv",sep=",",row.names=1)
taxonomy <- cbind(taxonomy, ASV = paste0("ASV", sprintf("%04d", 1:nrow(taxonomy))))
taxonomy=as.matrix(taxonomy)
TAX=tax_table(taxonomy)

metatable <- read.delim("microbiome/larvae/data_files/sample-metadata.txt") 
#View(metatable)
row.names(metatable) <- metatable[[1]]
metatable<- metatable[,(-1)]
META <- sample_data(metatable)

ps_l<- phyloseq(OTU, TAX, META)

# change taxonomy header 
colnames(tax_table(ps_l)) <- c(D0 = "Kingdom", D1 = "Phylum", D2 = "Class", 
                              D3 = "Order", D4 = "Family", D5 = "Genus", D6 = "Species", ASV = "ASV")

#The total number of ASVs in the whole dataset is 
length(taxa_names(ps_l))

#get rid of things we do not want
ps_larvae1 = subset_taxa(ps_l, Kingdom =="Bacteria")
ps_larvae1 <- prune_taxa(taxa_sums(ps_larvae1) > 0, ps_larvae1)
ps_larvae1<-subset_taxa(ps_larvae1, (Order!="Chloroplast"))
ps_larvae1<-subset_taxa(ps_larvae1, (Family!="Mitochondria"))
length(taxa_names(ps_larvae1))
#reduces total taxa numbers from 2017 to 1581 (minus 436)

# mean, max and min of sample read counts
smin <- min(sample_sums(ps_larvae1))
smean <- mean(sample_sums(ps_larvae1))
smax <- max(sample_sums(ps_larvae1))

# printing the results
cat("The minimum sample read count is:",smin)
cat("The average sample read count is:",smean)
cat("The maximum sample read count is:",smax)

#extract larvae samples
ps_larvae <- subset_samples(ps_larvae1, source == "larvae")

# mean, max and min of sample read counts
smin2 <- min(sample_sums(ps_larvae))
smean2 <- mean(sample_sums(ps_larvae))
smax2 <- max(sample_sums(ps_larvae))

# printing the results
cat("The minimum sample read count is:",smin2)
cat("The average sample read count is:",smean2)
cat("The maximum sample read count is:",smax2)
```

### Plot rarefaction curve
```{r rarefaction_curve2}
rare<- ps_larvae1
smin <- min(sample_sums(rare))
cat("The minimum sample read count is:",smin)

# rarefy
set.seed(42)
ps.rarefied = rarefy_even_depth(rare, rngseed=1, sample.size=7560, replace=F)
library(ranacapa)
cbPalette <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7","maroon","khaki4","coral4","blueviolet","chocolate3","gray48")

ggrare(ps.rarefied, step = 100, se = FALSE, color="treatment2")+ scale_colour_manual(values=cbPalette)+geom_line(size=1)+ theme_bw()+theme(axis.line = element_line(colour = "black"))+ theme(legend.title = element_blank(),
                legend.background = element_blank(),
                legend.key = element_blank()) + theme(text = element_text(size = 22))+theme(strip.text = element_text(face="bold", size=20))+ theme(axis.title.x=element_blank())+ggtitle("Rarefaction larvae samples")+theme(plot.title = element_text(hjust = 0.5, size=20, face="italic"))+theme(legend.text=element_text(size=17, face="italic"))
ggsave("microbiome/larvae/figures/rarefaction_larvae.png", height = 5, width = 12)
```
#The two samples which show much higher species richness are the samples "29.10-LG-III" and "29.10 LG-V"

## Alpha diversity larvae

```{r alpha_Shannon_larvae}
alpha<- ps_larvae
alpha <- subset_samples(alpha, treatment2 != "start_larvae")
#rarefy to even numbers
set.seed(1)  
alpha1_ra <- rarefy_even_depth(alpha,sample.size=7560, replace=FALSE, rngseed = 1) 
alpha1_ra<-prune_taxa(taxa_sums(alpha1_ra) > 0, alpha1_ra)

sample_data(alpha1_ra)$treatment_day<-factor(sample_data(alpha1_ra)$treatment_day,levels=c("three","four","six"), labels=c("Day three","Day four","Day six"))

sample_data(alpha1_ra)$treatment<-factor(sample_data(alpha1_ra)$treatment,levels=c("Hive","C","AG","LG","BB","LGBB"))
levels(sample_data(alpha1_ra)$treatment)

sigFunc = function(x){
  if(x < 0.001){"***"} 
  else if(x < 0.01){"**"}
  else if(x < 0.05){"*"}
  else{NA}}

alpha_meas = c("Shannon")
p2 <- plot_richness(alpha1_ra,"treatment",color="treatment", measures=alpha_meas)
p2$layers <- p2$layers[-1]
p2<-p2 + geom_boxplot(data=p2$data, aes(x=treatment, y=value),show_guide=FALSE, alpha=0.1)+geom_point(position = position_jitter(w = 0.2, h = 0.1),show_guide=FALSE) +
  geom_signif(comparisons=list(c("Hive", "C"), c("Hive", "AG"),c("Hive", "LG"),c("Hive", "BB"),c("Hive", "LGBB")),tip_length = 0.02,vjust=0.5, color=("grey20"),map_signif_level=sigFunc, y_position = c(4.2,4.8,5,5.4,6), margin_top = 0.1, textsize=6, test = wilcox.test)
p2 <- p2 + guides(fill=guide_legend(title=element_blank()))+ylim(-1,6)
p2 <- p2 + theme(plot.title = element_text(hjust = 0.5))+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"))
p2 <- p2+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+ theme(panel.background = element_blank())
p2 <- p2+theme(axis.line = element_line(colour = "black"))
p2<- p2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank()) +ylab("Shannon alpha diversity")
p2<-p2 + facet_wrap(~treatment_day, scales="free",nrow=2)
p2 <- p2+ theme(panel.border = element_blank())
p2<-p2 +theme(axis.text.y = element_text(size=16))+theme(axis.title.y = element_text(size=17,face="bold"))+theme(strip.text = element_text(face="bold", size=16, color="gray40"))+ theme(axis.title.x=element_blank())+theme(axis.text.x = element_text(size=16,angle=-50))+theme(strip.background =element_rect(fill="white", color="white"))
ggsave("microbiome/larvae/figures/alpha_Shannon_larvae_all.png", height = 6.5, width = 7.5)
```

```{r alpha_Shannon_larvae}
alpha<- ps_larvae
alpha <- subset_samples(alpha, treatment_day == "six")
#rarefy to even numbers
set.seed(1)  
alpha1_ra <- rarefy_even_depth(alpha,sample.size=7560, replace=FALSE, rngseed = 1) 
alpha1_ra<-prune_taxa(taxa_sums(alpha1_ra) > 0, alpha1_ra)
sample_data(alpha1_ra)$treatment<-factor(sample_data(alpha1_ra)$treatment,levels=c("Hive","C","AG","LG","BB","LGBB"))
levels(sample_data(alpha1_ra)$treatment)
sigFunc = function(x){
  if(x < 0.001){"***"} 
  else if(x < 0.01){"**"}
  else if(x < 0.05){"*"}
  else{NA}}
alpha_meas = c("Shannon")
p2 <- plot_richness(alpha1_ra,"treatment",color="treatment", measures=alpha_meas)
p2$layers <- p2$layers[-1]
p2<-p2 + geom_boxplot(data=p2$data, aes(x=treatment, y=value),show_guide=FALSE, alpha=0.1)+geom_point(position = position_jitter(w = 0.2, h = 0.1),show_guide=FALSE) +
  geom_signif(comparisons=list(c("Hive", "C"), c("Hive", "AG"),c("Hive", "LG"),c("Hive", "BB"),c("Hive", "LGBB")),tip_length = 0.02,vjust=0.5, color=("grey20"),map_signif_level=sigFunc, y_position = c(4.2,4.8,5,5.4,6), margin_top = 0.1, textsize=6, test = wilcox.test)
p2 <- p2 + guides(fill=guide_legend(title=element_blank()))+ylim(-1,6)
p2 <- p2 + theme(plot.title = element_text(hjust = 0.5))+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"))
p2 <- p2+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+ theme(panel.background = element_blank())
p2 <- p2+theme(axis.line = element_line(colour = "black"))
p2<- p2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank()) +ylab("Shannon alpha diversity")
p2 <- p2+ theme(panel.border = element_blank())+theme(strip.background =element_rect(fill="white", color="white"))+theme(strip.text = element_blank())
p2<-p2 +theme(axis.text.y = element_text(size=16))+theme(axis.title.y = element_text(size=17,face="bold"))+ theme(axis.title.x=element_blank())+theme(axis.text.x = element_text(size=16,angle=-50))
ggsave("microbiome/larvae/figures/alpha_Shannon_larvae_day_six.png", height = 3.5, width = 6.5)
```

### make statistical comparisons Shannon stats
```{r warning=FALSE}
alpha<- ps_larvae
alpha <- subset_samples(alpha, treatment2 != "start_larvae")
#rarefy to even numbers
set.seed(1)  
alpha1_ra <- rarefy_even_depth(alpha,sample.size=7560, replace=FALSE, rngseed = 1) 
alpha1_ra<-prune_taxa(taxa_sums(alpha1_ra) > 0, alpha1_ra)

results = estimate_richness(alpha1_ra, measures = 'Shannon')
d = sample_data(alpha1_ra)

# calculate wilcox-test between Hive control and treatments
# day three

Hive_3 = results[d[,'day_in_lab'] == 'Hive_three',]
C_3 = results[d[,'day_in_lab'] == 'C_three',]
AG_3 = results[d[,'day_in_lab'] == 'AG_three',]
LG_3 = results[d[,'day_in_lab'] == 'LG_three',]
BB_3 = results[d[,'day_in_lab'] == 'BB_three',]
LGBB_3 = results[d[,'day_in_lab'] == 'LGBB_three',]

wilcox.test(Hive_3, C_3)
wilcox.test(Hive_3, AG_3)
wilcox.test(Hive_3, LG_3)
wilcox.test(Hive_3, BB_3)
wilcox.test(Hive_3, LGBB_3)


pvalues<-c(0.0079,0.0079,0.841,0.0079,0.032)
p.adjust(pvalues,method="fdr")

# day four

Hive_4 = results[d[,'day_in_lab'] == 'Hive_four',]
C_4 = results[d[,'day_in_lab'] == 'C_four',]
AG_4 = results[d[,'day_in_lab'] == 'AG_four',]
LG_4 = results[d[,'day_in_lab'] == 'LG_four',]
BB_4 = results[d[,'day_in_lab'] == 'BB_four',]
LGBB_4 = results[d[,'day_in_lab'] == 'LGBB_four',]

wilcox.test(Hive_4, C_4)
wilcox.test(Hive_4, AG_4)
wilcox.test(Hive_4, LG_4)
wilcox.test(Hive_4, BB_4)
wilcox.test(Hive_4, LGBB_4)

pvalues<-c(0.4,1,0.1,0.2,0.1)
p.adjust(pvalues,method="fdr")

# day six

Hive_6 = results[d[,'day_in_lab'] == 'Hive_six',]
C_6 = results[d[,'day_in_lab'] == 'C_six',]
AG_6 = results[d[,'day_in_lab'] == 'AG_six',]
LG_6 = results[d[,'day_in_lab'] == 'LG_six',]
BB_6 = results[d[,'day_in_lab'] == 'BB_six',]
LGBB_6 = results[d[,'day_in_lab'] == 'LGBB_six',]

wilcox.test(Hive_6, C_6)
wilcox.test(Hive_6, AG_6)
wilcox.test(Hive_6, LG_6)
wilcox.test(Hive_6, BB_6)
wilcox.test(Hive_6, LGBB_6)

pvalues<-c(0.004,1,1,0.132,0.699)
p.adjust(pvalues,method="fdr")

#statistics

alpha<- ps_larvae
set.seed(1)  
alpha <- rarefy_even_depth(alpha,sample.size=7560, replace=FALSE, rngseed = 1) 

alpha_day_three <- subset_samples(alpha, treatment_day == "three")
alpha_day_three<-prune_taxa(taxa_sums(alpha_day_three) > 0, alpha_day_three)
rich = estimate_richness(alpha_day_three)
pairwise.wilcox.test(rich$Observed,sample_data(alpha_day_three)$treatment,p.adjust.method ="fdr")
pairwise.wilcox.test(rich$Shannon,sample_data(alpha_day_three)$treatment,p.adjust.method ="fdr")

alpha_day_four <- subset_samples(alpha, treatment_day == "four")
alpha_day_four<-prune_taxa(taxa_sums(alpha_day_four) > 0, alpha_day_four)
rich2 = estimate_richness(alpha_day_four)
pairwise.wilcox.test(rich2$Observed,sample_data(alpha_day_four)$treatment,p.adjust.method ="fdr")
pairwise.wilcox.test(rich2$Shannon,sample_data(alpha_day_four)$treatment,p.adjust.method ="fdr")

alpha_day_six <- subset_samples(alpha, treatment_day == "six")
alpha_day_six<-prune_taxa(taxa_sums(alpha_day_six) > 0, alpha_day_six)
rich3 = estimate_richness(alpha_day_six)
pairwise.wilcox.test(rich3$Observed,sample_data(alpha_day_six)$treatment,p.adjust.method ="fdr")
pairwise.wilcox.test(rich3$Shannon,sample_data(alpha_day_six)$treatment,p.adjust.method ="fdr")
```


```{r alpha_Observed_larvae_all}
alpha<- ps_larvae
alpha <- subset_samples(alpha, treatment2 != "start_larvae")
#rarefy to even numbers
set.seed(1)  
alpha1_ra <- rarefy_even_depth(alpha,sample.size=7560, replace=FALSE, rngseed = 1) 
alpha1_ra<-prune_taxa(taxa_sums(alpha1_ra) > 0, alpha1_ra)

sample_data(alpha1_ra)$treatment_day<-factor(sample_data(alpha1_ra)$treatment_day,levels=c("three","four","six"), labels=c("Day three","Day four","Day six"))

sample_data(alpha1_ra)$treatment<-factor(sample_data(alpha1_ra)$treatment,levels=c("Hive","C","AG","LG","BB","LGBB"))
                                         #, labels=c("Hive control","RJ","RJ \n +adult gut", "RJ \n +larvae gut", "RJ \n +bee bread","RJ \n +larvae gut \n +bee bread"))
levels(sample_data(alpha1_ra)$treatment)

sigFunc = function(x){
  if(x < 0.001){"***"} 
  else if(x < 0.01){"**"}
  else if(x < 0.05){"*"}
  else{NA}}

alpha_meas = c("Observed")
p2 <- plot_richness(alpha1_ra,"treatment",color="treatment", measures=alpha_meas)
p2$layers <- p2$layers[-1]
p2<-p2 + geom_boxplot(data=p2$data, aes(x=treatment, y=value),show_guide=FALSE, alpha=0.1)+geom_point(position = position_jitter(w = 0.2, h = 0.1),show_guide=FALSE) +
  geom_signif(comparisons=list(c("Hive", "C"), c("Hive", "AG"),c("Hive", "LG"),c("Hive", "BB"),c("Hive", "LGBB")),tip_length = 0.02,vjust=0.5, color=("grey20"),map_signif_level=sigFunc, y_position = c(120,135,140,150,150), margin_top = 0.1, textsize=6, test = wilcox.test)
p2 <- p2 + guides(fill=guide_legend(title=element_blank()))+ylim(0,155)
p2 <- p2 + theme(plot.title = element_text(hjust = 0.5))+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"))
p2 <- p2+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+ theme(panel.background = element_blank())
p2 <- p2+theme(axis.line = element_line(colour = "black"))
p2<- p2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank()) +ylab("Observed species")
p2<-p2 + facet_wrap(~treatment_day, scales="free",nrow=2)
p2 <- p2+ theme(panel.border = element_blank())
p2<-p2 +theme(axis.text.y = element_text(size=16))+theme(axis.title.y = element_text(size=17,face="bold"))+theme(strip.text = element_text(face="bold", size=16, color="gray40"))+ theme(axis.title.x=element_blank())+theme(axis.text.x = element_text(size=16,angle=-50))+theme(strip.background =element_rect(fill="white", color="white"))
ggsave("microbiome/larvae/figures/alpha_Observed_larvae_all.png", height = 6.5, width = 7.5)
```

```{r alpha_Observed_larvae, fig.heigt=13, fig.width=8}
alpha<- ps_larvae
alpha <- subset_samples(alpha, treatment_day == "six")
#rarefy to even numbers
set.seed(1)  
alpha1_ra <- rarefy_even_depth(alpha,sample.size=7560, replace=FALSE, rngseed = 1) 
alpha1_ra<-prune_taxa(taxa_sums(alpha1_ra) > 0, alpha1_ra)

sample_data(alpha1_ra)$treatment<-factor(sample_data(alpha1_ra)$treatment,levels=c("Hive","C","AG","LG","BB","LGBB"))

levels(sample_data(alpha1_ra)$treatment)

sigFunc = function(x){
  if(x < 0.001){"***"} 
  else if(x < 0.01){"**"}
  else if(x < 0.05){"*"}
  else{NA}}

alpha_meas = c("Observed")
p2 <- plot_richness(alpha1_ra,"treatment",color="treatment", measures=alpha_meas)
p2$layers <- p2$layers[-1]
p2<-p2 + geom_boxplot(data=p2$data, aes(x=treatment, y=value),show_guide=FALSE, alpha=0.1)+geom_point(position = position_jitter(w = 0.2, h = 0.1),show_guide=FALSE) +
  geom_signif(comparisons=list(c("Hive", "C"), c("Hive", "AG"),c("Hive", "LG"),c("Hive", "BB"),c("Hive", "LGBB")),tip_length = 0.02,vjust=0.5, color=("grey20"),map_signif_level=sigFunc, y_position = c(60,70,70,70,70), margin_top = 0.1, textsize=6, test = wilcox.test)
p2 <- p2 + guides(fill=guide_legend(title=element_blank()))+ylim(0,80)
p2 <- p2 + theme(plot.title = element_text(hjust = 0.5))+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"))
p2 <- p2+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+ theme(panel.background = element_blank())
p2 <- p2+theme(axis.line = element_line(colour = "black"))
p2<- p2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank()) +ylab("Observed species")
p2 <- p2+ theme(panel.border = element_blank())
Observed_larvae<-p2 +theme(axis.text.y = element_text(size=15))+theme(axis.title.y = element_text(size=16))+theme(strip.text = element_blank())+ theme(axis.title.x=element_blank())+theme(axis.text.x = element_text(size=15,angle=-50))+theme(strip.background =element_rect(fill="white", color="white"))
Observed_larvae<-Observed_larvae+ theme(axis.ticks.x = element_blank())+theme(axis.text.x = element_blank())
ggsave("microbiome/larvae/figures/alpha_Observed_larvae_day_six.png", height = 3.5, width = 6.5)
```

### make statistical comparisons Observed stats
```{r warning=FALSE}
alpha<- ps_larvae
alpha <- subset_samples(alpha, treatment2 != "start_larvae")
#rarefy to even numbers
set.seed(1)  
alpha1_ra <- rarefy_even_depth(alpha,sample.size=7560, replace=FALSE, rngseed = 1) 
alpha1_ra<-prune_taxa(taxa_sums(alpha1_ra) > 0, alpha1_ra)

results = estimate_richness(alpha1_ra, measures = 'Observed')
d = sample_data(alpha1_ra)

# calculate wilcox-test between Hive control and treatments
# day three

Hive_3 = results[d[,'day_in_lab'] == 'Hive_three',]
C_3 = results[d[,'day_in_lab'] == 'C_three',]
AG_3 = results[d[,'day_in_lab'] == 'AG_three',]
LG_3 = results[d[,'day_in_lab'] == 'LG_three',]
BB_3 = results[d[,'day_in_lab'] == 'BB_three',]
LGBB_3 = results[d[,'day_in_lab'] == 'LGBB_three',]

wilcox.test(Hive_3, C_3)
wilcox.test(Hive_3, AG_3)
wilcox.test(Hive_3, LG_3)
wilcox.test(Hive_3, BB_3)
wilcox.test(Hive_3, LGBB_3)


pvalues<-c(0.008,0.008,0.0159,0.008,0.091)
p.adjust(pvalues,method="fdr")

# day four

Hive_4 = results[d[,'day_in_lab'] == 'Hive_four',]
C_4 = results[d[,'day_in_lab'] == 'C_four',]
AG_4 = results[d[,'day_in_lab'] == 'AG_four',]
LG_4 = results[d[,'day_in_lab'] == 'LG_four',]
BB_4 = results[d[,'day_in_lab'] == 'BB_four',]
LGBB_4 = results[d[,'day_in_lab'] == 'LGBB_four',]

wilcox.test(Hive_4, C_4)
wilcox.test(Hive_4, AG_4)
wilcox.test(Hive_4, LG_4)
wilcox.test(Hive_4, BB_4)
wilcox.test(Hive_4, LGBB_4)

pvalues<-c(0.825,0.1,0.4,0.7,0.4)
p.adjust(pvalues,method="fdr")


# day six

Hive_6 = results[d[,'day_in_lab'] == 'Hive_six',]
C_6 = results[d[,'day_in_lab'] == 'C_six',]
AG_6 = results[d[,'day_in_lab'] == 'AG_six',]
LG_6 = results[d[,'day_in_lab'] == 'LG_six',]
BB_6 = results[d[,'day_in_lab'] == 'BB_six',]
LGBB_6 = results[d[,'day_in_lab'] == 'LGBB_six',]

wilcox.test(Hive_6, C_6)
wilcox.test(Hive_6, AG_6)
wilcox.test(Hive_6, LG_6)
wilcox.test(Hive_6, BB_6)
wilcox.test(Hive_6, LGBB_6)

pvalues<-c(0.0043,0.054,0.568,0.002,0.145)
p.adjust(pvalues,method="fdr")
```


### ordination plots
```{r NMDS_larvae}
#NMDS on Bray-Curtis
ord2<- ps_larvae
ord2 = subset_samples(ord2, treatment_day== "six")
ord2 <- transform_sample_counts(ord2, function(OTU) {OTU / sum(OTU)})
sample_data(ord2)$treatment<-factor(sample_data(ord2)$treatment,levels=c("Hive","C","AG","LG","BB","LGBB"))
levels(sample_data(ord2)$treatment)

set.seed(1)
dist<-phyloseq::distance(ord2, method="bray") 
ordination = ordinate(ord2, method="NMDS", distance=dist)
cat("stress is:", ordination$stress)
p2 <- plot_ordination(ord2, ordination, color="treatment") +geom_point(size=2.5)
p2 <- p2 + guides(fill=guide_legend(title=element_blank()))
p2 <- p2 + theme(plot.title = element_text(hjust = 0.5))+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"),guide = guide_legend(override.aes = list(shape = c(rep(15, 6)),size=9,alpha=0.3)))
p2 <- p2+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+ theme(panel.background = element_blank())
p2 <- p2+theme(axis.line = element_line(colour = "black"))
p2<- p2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank())
p2 <- p2+ theme(panel.border = element_blank())
p2<-p2 + theme(text = element_text(size = 16))+theme(strip.text = element_blank())
p2<-p2+theme(legend.position="bottom")+theme(strip.background =element_rect(fill="white", color="white"))
NMDS<-p2+stat_ellipse(aes(group = treatment),size=1.5, alpha=0.5)+ theme(panel.spacing.x=unit(5.5, "lines"))+ theme(axis.title.y = element_text(margin = margin(t = 0, r = 0, b = 0, l = 20)))+ theme(legend.text = element_text(face="bold", size=16))
ggsave("microbiome/larvae/figures/NMDS_larvae_day_six.png", height = 5.5, width = 7.5)
```

```{r PCoA_larva}
#PCoA on Bray-Curtis
ord2<- ps_larvae
ord2 = subset_samples(ord2, treatment_day== "six")
ord2 <- transform_sample_counts(ord2, function(OTU) {OTU / sum(OTU)})
sample_data(ord2)$treatment<-factor(sample_data(ord2)$treatment,levels=c("Hive","C","AG","LG","BB","LGBB"))
levels(sample_data(ord2)$treatment)
set.seed(1)
dist<-phyloseq::distance(ord2, method="bray") 
ordination = ordinate(ord2, method="PCoA", distance=dist)
p2 <- plot_ordination(ord2, ordination, color="treatment") +geom_point(size=2.5)
p2 <- p2 + guides(fill=guide_legend(title=element_blank()))
p2 <- p2 + theme(plot.title = element_text(hjust = 0.5))+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"),guide = guide_legend(override.aes = list(shape = c(rep(15, 6)),size=9,alpha=0.3)))
p2 <- p2+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+ theme(panel.background = element_blank())
p2 <- p2+theme(axis.line = element_line(colour = "black"))
p2<- p2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank())
p2 <- p2+ theme(panel.border = element_blank())
p2<-p2 + theme(text = element_text(size = 16))+theme(strip.text = element_blank())
p2<-p2+theme(legend.position="bottom")+theme(strip.background =element_rect(fill="white", color="white"))
PCoA_larvae<-p2+stat_ellipse(aes(group = treatment),size=1.5, alpha=0.5)+ theme(panel.spacing.x=unit(5.5, "lines"))+ theme(axis.title.y = element_text(margin = margin(t = 0, r = 0, b = 0, l = 20)))+ theme(legend.text = element_text(face="bold", size=17,color="gray9"))
ggsave("microbiome/larvae/figures/PCoA_day_six.png", height = 5.5, width = 8)
```

### Beta diversity stats

#### test difference between treatments and hive control

```{r}
#day 3
set.seed(42)
compare=ps_larvae
compare1 <- transform_sample_counts(compare, function(OTU) {OTU / sum(OTU)})

day3 = subset_samples(compare1, treatment_day== "three")
day4 = subset_samples(compare1, treatment_day== "four")
day6 = subset_samples(compare1, treatment_day== "six")

df = as(sample_data(day3), "data.frame")
d = distance(day3, "bray")
adonis = adonis(d ~ treatment, df, perm=999)
adonis

df = as(sample_data(day4), "data.frame")
d = distance(day4, "bray")
adonis = adonis(d ~ treatment, df, perm=999)
adonis

df = as(sample_data(day6), "data.frame")
d = distance(day6, "bray")
adonis = adonis(d ~ treatment, df, perm=999)
adonis

subs <- subset_samples(compare1, day_in_lab %in% c("Hive_three", "C_three"))
subs <- prune_taxa(taxa_sums(subs) > 0, subs)
metadata <- as(sample_data(subs), "data.frame")
adonis(distance(subs, method="bray") ~ treatment,data = metadata, perm=999)

subs1 <- subset_samples(compare1, day_in_lab %in% c("Hive_three", "AG_three"))
subs1 <- prune_taxa(taxa_sums(subs1) > 0, subs1)
metadata <- as(sample_data(subs1), "data.frame")
adonis(distance(subs1, method="bray") ~ treatment,data = metadata, perm=999)

subs2 <- subset_samples(compare1, day_in_lab %in% c("Hive_three", "LG_three"))
subs2 <- prune_taxa(taxa_sums(subs2) > 0, subs2)
metadata <- as(sample_data(subs2), "data.frame")
adonis(distance(subs2, method="bray") ~ treatment,data = metadata, perm=999)

subs <- subset_samples(compare1, day_in_lab %in% c("Hive_three", "BB_three"))
subs <- prune_taxa(taxa_sums(subs) > 0, subs)
metadata <- as(sample_data(subs), "data.frame")
adonis(distance(subs, method="bray") ~ treatment,data = metadata, perm=999)

subs3 <- subset_samples(compare1, day_in_lab %in% c("Hive_three", "LGBB_three"))
subs3 <- prune_taxa(taxa_sums(subs3) > 0, subs3)
metadata <- as(sample_data(subs3), "data.frame")
adonis(distance(subs3, method="bray") ~ treatment,data = metadata, perm=999)

pvalues<-c(0.009,0.012,0.016,0.009,0.009)
p.adjust(pvalues,method="fdr")
```

```{r}
#day 4
set.seed(42)
compare=ps_larvae
compare1 <- transform_sample_counts(compare, function(OTU) {OTU / sum(OTU)})

subs <- subset_samples(compare1, day_in_lab %in% c("Hive_four", "C_four"))
subs <- prune_taxa(taxa_sums(subs) > 0, subs)
metadata <- as(sample_data(subs), "data.frame")
adonis(distance(subs, method="bray") ~ treatment,data = metadata, perm=999)

subs1 <- subset_samples(compare1, day_in_lab %in% c("Hive_four", "AG_four"))
subs1 <- prune_taxa(taxa_sums(subs1) > 0, subs1)
metadata <- as(sample_data(subs1), "data.frame")
adonis(distance(subs1, method="bray") ~ treatment,data = metadata, perm=999)

subs2 <- subset_samples(compare1, day_in_lab %in% c("Hive_four", "LG_four"))
subs2 <- prune_taxa(taxa_sums(subs2) > 0, subs2)
metadata <- as(sample_data(subs2), "data.frame")
adonis(distance(subs2, method="bray") ~ treatment,data = metadata, perm=999)

subs <- subset_samples(compare1, day_in_lab %in% c("Hive_six", "BB_six"))
subs <- prune_taxa(taxa_sums(subs) > 0, subs)
metadata <- as(sample_data(subs), "data.frame")
adonis(distance(subs, method="bray") ~ treatment,data = metadata, perm=999)

subs3 <- subset_samples(compare1, day_in_lab %in% c("Hive_four", "LGBB_four"))
subs3 <- prune_taxa(taxa_sums(subs3) > 0, subs3)
metadata <- as(sample_data(subs3), "data.frame")
adonis(distance(subs3, method="bray") ~ treatment,data = metadata, perm=999)

pvalues<-c(0.2,0.1,0.1,0.002,0.1)
p.adjust(pvalues,method="fdr")
```

```{r}
#day 6
set.seed(42)
compare=ps_larvae
compare1 <- transform_sample_counts(compare, function(OTU) {OTU / sum(OTU)})

subs <- subset_samples(compare1, day_in_lab %in% c("Hive_six", "C_six"))
subs <- prune_taxa(taxa_sums(subs) > 0, subs)
metadata <- as(sample_data(subs), "data.frame")
adonis(distance(subs, method="bray") ~ treatment,data = metadata, perm=999)

subs1 <- subset_samples(compare1, day_in_lab %in% c("Hive_six", "AG_six"))
subs1 <- prune_taxa(taxa_sums(subs1) > 0, subs1)
metadata <- as(sample_data(subs1), "data.frame")
adonis(distance(subs1, method="bray") ~ treatment,data = metadata, perm=999)

subs2 <- subset_samples(compare1, day_in_lab %in% c("Hive_six", "LG_six"))
subs2 <- prune_taxa(taxa_sums(subs2) > 0, subs2)
metadata <- as(sample_data(subs2), "data.frame")
adonis(distance(subs2, method="bray") ~ treatment,data = metadata, perm=999)

subs <- subset_samples(compare1, day_in_lab %in% c("Hive_six", "BB_six"))
subs <- prune_taxa(taxa_sums(subs) > 0, subs)
metadata <- as(sample_data(subs), "data.frame")
adonis(distance(subs, method="bray") ~ treatment,data = metadata, perm=999)

subs3 <- subset_samples(compare1, day_in_lab %in% c("Hive_six", "LGBB_six"))
subs3 <- prune_taxa(taxa_sums(subs3) > 0, subs3)
metadata <- as(sample_data(subs3), "data.frame")
adonis(distance(subs3, method="bray") ~ treatment,data = metadata, perm=999)

pvalues<-c(0.004,0.009,0.004,0.003,0.003)
p.adjust(pvalues,method="fdr")
```

### Larvae Taxonomy plots and stats

```{r individual_larva}
tax <-ps_larvae
tax = subset_samples(tax, treatment_day == "six")
tax <- prune_taxa(taxa_sums(tax) > 0, tax)

ps_bdiv2 <- transform_sample_counts(tax, function(OTU) {OTU / sum(OTU)})
tax_all2<- ps_bdiv2
ps_pd2 <- tax_glom(tax_all2, taxrank = 'Genus')
low1pc_reads <- max(taxa_sums(ps_pd2)) * 1 / 100
low1pc_indices <- which(taxa_sums(ps_pd2) < low1pc_reads)
length(low1pc_indices)
taxa_names(ps_pd2)[low1pc_indices[1]] <- "other"
taxa3 <- merge_taxa(ps_pd2, low1pc_indices, "other")
tax_table(taxa3)["other", ] <- c("other")

#build a custom color palette for phyloseq object
getPalette = colorRampPalette(brewer.pal(9, "Set3"))
speciesList = unique(tax_table(taxa3)[,"Genus"])
speciesPalette = getPalette(length(speciesList))
names(speciesPalette) = speciesList

ps_df <- psmelt(taxa3)
ps_df_sum <- ps_df %>%
  group_by(Sample,treatment,Genus) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(Sample,treatment) %>%
  mutate(relative_abundance = Abundance / sum(Abundance)) %>%
  ungroup() %>%
  mutate(Genus = fct_reorder(Genus, Abundance, .fun = sum, .desc = TRUE))
levels(ps_df_sum$Genus)

neworder <- c("Hive","C","AG","LG","BB","LGBB")
change <- c("Hive\n  ","C\n  ","AG\n  ","LG\n  ","BB\n  ","LGBB\n  ")
ps_df_sum2 <- arrange(transform(ps_df_sum,treatment=factor(treatment,levels=neworder,labels=change)),treatment)

p<-ggplot(data = ps_df_sum2, aes(x = Sample, y = relative_abundance, fill = Genus)) +
  geom_bar(stat = "identity")+ scale_fill_manual(values= speciesPalette)+ scale_y_continuous(expand = c(0,0))+ylab("relative abundance")
p<-p+ theme(legend.text=element_text(size=10.5, face = "italic"))+theme(legend.key = element_rect(color = NA, fill = NA),legend.key.size = unit(0.5, "cm"))+theme(legend.title = element_blank())+theme(legend.position="bottom")
taxa2<-p+theme(axis.title.y = element_text(size=12, face="bold"))+theme(axis.text.y = element_text(size=11, face="bold"))+theme(axis.text.x = element_text(size=7, face="bold",vjust=0.5,angle = 90))+theme(axis.title.x = element_blank())+theme(strip.background =element_rect(fill="white", color="white"))+theme(strip.text = element_text(size=10, face="bold",vjust = -5))#+theme(axis.text.x =element_blank())
taxa2<-taxa2+guides(fill=guide_legend(nrow=3))  
taxa2<-taxa2+facet_grid(~treatment, scale='free_x', space = "free_x")
taxa2<-taxa2+theme(axis.text.x = element_blank(), axis.ticks.x=element_blank())
#taxa2+facet_grid(~treatment, scale='free_x', space = "free_x")
ggsave("microbiome/larvae/figures/taxa_larvae_individuals.png", height = 3.5, width = 7)
```

```{r all_taxa_plot}
tax <-ps_larvae
tax = subset_samples(tax, treatment2 != "start_larvae")
ps_bdiv2 <- transform_sample_counts(tax, function(OTU) {OTU / sum(OTU)})
tax_all2<- ps_bdiv2
ps_pd2 <- tax_glom(tax_all2, taxrank = 'Genus')

low1pc_reads <- max(taxa_sums(ps_pd2)) * 1 / 100
low1pc_indices <- which(taxa_sums(ps_pd2) < low1pc_reads)
length(low1pc_indices)
taxa_names(ps_pd2)[low1pc_indices[1]] <- "other"
taxa3 <- merge_taxa(ps_pd2, low1pc_indices, "other")
tax_table(taxa3)["other", ] <- c("other")

#build a custom color palette for phyloseq object
getPalette = colorRampPalette(brewer.pal(12, "Set3"))
speciesList = unique(tax_table(taxa3)[,"Genus"])
speciesPalette = getPalette(length(speciesList))
names(speciesPalette) = speciesList

ps_df <- psmelt(taxa3)
ps_df_sum <- ps_df %>%
  group_by(treatment2,Genus,treatment_day) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(treatment2,treatment_day) %>%
  mutate(relative_abundance = Abundance / sum(Abundance)) %>%
  ungroup() %>%
  mutate(Genus = fct_reorder(Genus, Abundance, .fun = sum, .desc = TRUE))
levels(ps_df_sum$Genus)

neworder <- c("Hive","C","AG","LG","BB","LGBB")
change <- c("Hive\n  ","C\n  ","AG\n  ","LG\n  ","BB\n  ","LGBB\n  ")
ps_df_sum2 <- arrange(transform(ps_df_sum,treatment2=factor(treatment2,levels=neworder,labels=change)),treatment2)

neworder2 <- c("three","four","six")
change2 <- c("day three","day four","day six")
ps_df_sum3 <- arrange(transform(ps_df_sum2,treatment_day=factor(treatment_day,levels=neworder2,labels=change2)),treatment_day)

p<-ggplot(data = ps_df_sum3, aes(x = treatment2, y = relative_abundance, fill = Genus)) +
  geom_bar(stat = "identity")+ scale_fill_manual(values= speciesPalette)+ scale_y_continuous(expand = c(0,0))+ylab("relative abundance")
p<-p+ theme(legend.position="bottom")
p<-p + theme(strip.text = element_text(size=13,face="bold",color="grey40"))
p<-p+ theme(legend.text=element_text(size=12, face = "italic"))+theme(legend.key = element_rect(color = NA, fill = NA),
                                                     legend.key.size = unit(0.6, "cm"))+theme(legend.title = element_blank())
p<-p + facet_wrap( ~treatment_day, scales="free_x")
taxa2<-p+theme(axis.title.y = element_text(size=12, face="bold"))+theme(axis.text.y = element_text(size=11, face="bold"))+theme(axis.text.x = element_text(size=11, face="bold",vjust=0.5))+theme(axis.title.x = element_blank())+theme(strip.background =element_rect(fill="white", color="white"))#+theme(axis.text.x =element_blank())
ggsave("microbiome/larvae/figures/taxa_larvae_all.png", height = 5, width = 10)
```

```{r other_taxa_plot}
tax <-ps_larvae1
tax = subset_samples(tax, treatment == "AG_inoculum"|treatment =="LG_inoculum"|treatment=="BB_food")

ps_bdiv2 <- transform_sample_counts(tax, function(OTU) {OTU / sum(OTU)})
tax_all2<- ps_bdiv2
ps_pd2 <- tax_glom(tax_all2, taxrank = 'Genus')

low1pc_reads <- max(taxa_sums(ps_pd2)) * 1 / 10
low1pc_indices <- which(taxa_sums(ps_pd2) < low1pc_reads)
length(low1pc_indices)
taxa_names(ps_pd2)[low1pc_indices[1]] <- "other"
taxa3 <- merge_taxa(ps_pd2, low1pc_indices, "other")
tax_table(taxa3)["other", ] <- c("other")

#build a custom color palette for phyloseq object
getPalette = colorRampPalette(brewer.pal(9, "Set3"))
speciesList = unique(tax_table(taxa3)[,"Genus"])
speciesPalette = getPalette(length(speciesList))
names(speciesPalette) = speciesList

ps_df <- psmelt(taxa3)
ps_df_sum <- ps_df %>%
  group_by(treatment2,Genus) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(treatment2) %>%
  mutate(relative_abundance = Abundance / sum(Abundance)) %>%
  ungroup() %>%
  mutate(Genus = fct_reorder(Genus, Abundance, .fun = sum, .desc = TRUE))
levels(ps_df_sum$Genus)

neworder <- c("AG_inoculum","BB_food","LG_inoculum")
change <- c("AG\ninoculum","BB\nfood","LG\ninoculum")
ps_df_sum2 <- arrange(transform(ps_df_sum,treatment2=factor(treatment2,levels=neworder,labels=change)),treatment2)

p<-ggplot(data = ps_df_sum2, aes(x = treatment2, y = relative_abundance, fill = Genus)) +
  geom_bar(stat = "identity")+ scale_fill_manual(values= speciesPalette)+ scale_y_continuous(expand = c(0,0))+ylab("relative abundance")
p<-p+ theme(legend.text=element_text(size=10, face = "italic"))+theme(legend.key = element_rect(color = NA, fill = NA),
                                                     legend.key.size = unit(0.6, "cm"))+theme(legend.title = element_blank())
taxa2<-p+theme(axis.title.y = element_text(size=12, face="bold"))+theme(axis.text.y = element_text(size=11, face="bold"))+theme(axis.text.x = element_text(size=11, face="bold",vjust=0.5))+theme(axis.title.x = element_blank())
ggsave("microbiome/larvae/figures/taxa_inoculum.png", height = 3.1, width = 4.5)
```


```{r start_taxa}
tax <-ps_larvae1
tax = subset_samples(tax, treatment2 == "start_larvae")

ps_bdiv2 <- transform_sample_counts(tax, function(OTU) {OTU / sum(OTU)})
tax_all2<- ps_bdiv2
ps_pd2 <- tax_glom(tax_all2, taxrank = 'Genus')

low1pc_reads <- max(taxa_sums(ps_pd2)) * 1 / 10
low1pc_indices <- which(taxa_sums(ps_pd2) < low1pc_reads)
length(low1pc_indices)
taxa_names(ps_pd2)[low1pc_indices[1]] <- "other"
taxa3 <- merge_taxa(ps_pd2, low1pc_indices, "other")
tax_table(taxa3)["other", ] <- c("other")

ps_df <- psmelt(taxa3)
ps_df_sum <- ps_df %>%
  group_by(treatment2,Genus) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(treatment2) %>%
  mutate(relative_abundance = Abundance / sum(Abundance)) %>%
  ungroup() %>%
  mutate(Genus = fct_reorder(Genus, Abundance, .fun = sum, .desc = TRUE))
levels(ps_df_sum$Genus)

neworder <- c("start_larvae")
change <- c("Larvae\nstart day")
ps_df_sum2 <- arrange(transform(ps_df_sum,treatment2=factor(treatment2,levels=neworder,labels=change)),treatment2)

p<-ggplot(data = ps_df_sum2, aes(x = treatment2, y = relative_abundance, fill = Genus)) +
  geom_bar(stat = "identity")+ scale_fill_manual(values= speciesPalette)+ scale_y_continuous(expand = c(0,0))+ylab("relative abundance")
p<-p+ theme(legend.text=element_text(size=10, face = "italic"))+theme(legend.key = element_rect(color = NA, fill = NA),
                                                     legend.key.size = unit(0.6, "cm"))+theme(legend.title = element_blank())
taxa2<-p+theme(axis.title.y = element_text(size=12, face="bold"))+theme(axis.text.y = element_text(size=11, face="bold"))+theme(axis.text.x = element_text(size=11, face="bold",vjust=0.5))+theme(axis.title.x = element_blank())
ggsave("microbiome/larvae/figures/start_larvae.png", height = 3, width = 3)
```

### test spread of variance of groups
```{r betadisp_larva}
set.seed(53)
test<-ps_larvae
test2 <- transform_sample_counts(test, function(OTU) {OTU / sum(OTU)})

cycle3 = subset_samples(test2, treatment_day == "three")
cycle3 <- prune_taxa(taxa_sums(cycle3) > 0, cycle3)

cycle4 = subset_samples(test2, treatment_day == "four")
cycle4 <- prune_taxa(taxa_sums(cycle4) > 0, cycle4)

cycle6 = subset_samples(test2, treatment_day == "six")
cycle6 <- prune_taxa(taxa_sums(cycle6) > 0, cycle6)

#testing treatment
set.seed(53)
d_cycle3 = distance(cycle3, "bray")
df_cycle3 = as(sample_data(cycle3), "data.frame")
df_cycle3$treatment <- factor(df_cycle3$treatment , levels=c("Hive","C","AG","LG","BB","LGBB"))
groups <- df_cycle3[["treatment"]]
beta <- betadisper(d_cycle3, df_cycle3$treatment)
permutest(beta,pairwise = TRUE, permutations = 999)
anova(betadisper(d_cycle3, groups))
mod.HSD <- TukeyHSD(beta,conf.level = 0.95)
mod.HSD
plot(betadisper(d_cycle3, groups),main="Day three",lwd=2, label = TRUE,label.cex = 0.7,col=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","hotpink"))
boxplot(betadisper(d_cycle3, groups),main="Day three",border=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","hotpink"),xlab="")

set.seed(53)
d_cycle4 = distance(cycle4, "bray")
df_cycle4 = as(sample_data(cycle4), "data.frame")
df_cycle4$treatment <- factor(df_cycle4$treatment, levels=c("Hive","C","AG","LG","BB","LGBB"))
groups <- df_cycle4[["treatment"]]
beta <- betadisper(d_cycle4, df_cycle4$treatment)
permutest(beta,pairwise = TRUE, permutations = 999)
anova(betadisper(d_cycle4, groups))
mod.HSD <- TukeyHSD(beta,conf.level = 0.95)
mod.HSD
plot(betadisper(d_cycle4, groups),main="Day four",lwd=2, label = TRUE,label.cex = 0.7,col=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","hotpink"))
boxplot(betadisper(d_cycle4, groups),main="Day four",border=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","hotpink"),xlab="")

set.seed(53)
d_cycle6 = distance(cycle6, "bray")
df_cycle6 = as(sample_data(cycle6), "data.frame")
df_cycle6$treatment <- factor(df_cycle6$treatment , levels=c("Hive","C","AG","LG","BB","LGBB"))
groups <- df_cycle6[["treatment"]]
beta <- betadisper(d_cycle6, df_cycle6$treatment)
permutest(beta,pairwise = TRUE, permutations = 999)
anova(betadisper(d_cycle6, groups))
mod.HSD <- TukeyHSD(beta,conf.level = 0.95)
mod.HSD
plot(betadisper(d_cycle6, groups),main="Larvae day six",lwd=2, label = TRUE,label.cex = 0.7,col=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","hotpink"))
boxplot(betadisper(d_cycle6, groups),main="Larvae day six",border=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","hotpink"),xlab="")
```

### make taxa plot with all samples - larvae and adults

```{r merge}
ps_larvae_merge<-subset_samples(ps_larvae, treatment_day == "six")
ps_adults_merge<-ps_adults

ps_both<-merge_phyloseq(ps_adults_merge, ps_larvae_merge)

ps_both<-subset_samples(ps_both, treatment == "Hive"|treatment == "AG"|treatment == "LG"|treatment == "BB"|treatment == "LGBB"|treatment == "C")
```

```{r merge_taxa}
library(pals)
tax <-ps_both
ps_bdiv2 <- transform_sample_counts(tax, function(OTU) {OTU / sum(OTU)})
tax_all2<- ps_bdiv2
ps_pd2 <- tax_glom(tax_all2, taxrank = 'Genus')

low1pc_reads <- max(taxa_sums(ps_pd2)) * 1 / 100
low1pc_indices <- which(taxa_sums(ps_pd2) < low1pc_reads)
length(low1pc_indices)
taxa_names(ps_pd2)[low1pc_indices[1]] <- "other"
taxa3 <- merge_taxa(ps_pd2, low1pc_indices, "other")
tax_table(taxa3)["other", ] <- c("other")

#build a custom color palette for phyloseq object
getPalette = colorRampPalette(brewer.pal(15, "Paired"))
speciesList = unique(tax_table(taxa3)[,"Genus"])
speciesPalette = getPalette(length(speciesList))
names(speciesPalette) = speciesList

ps_df <- psmelt(taxa3)
ps_df_sum <- ps_df %>%
  group_by(Sample,treatment,Genus,source) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(Sample,treatment,source) %>%
  mutate(relative_abundance = Abundance / sum(Abundance)) %>%
  ungroup() %>%
  mutate(Genus = fct_reorder(Genus, Abundance, .fun = sum, .desc = TRUE))
levels(ps_df_sum$Genus)

neworder <- c("Hive","C","AG","LG","BB","LGBB")
change <- c("Hive\n  ","C\n  ","AG\n  ","LG\n  ","BB\n  ","LGBB\n  ")
ps_df_sum2 <- arrange(transform(ps_df_sum,treatment=factor(treatment,levels=neworder,labels=change)),treatment)

neworder2 <- c("larvae","adults")
ps_df_sum3 <- arrange(transform(ps_df_sum2,source=factor(source,levels=neworder2)),source)

p<-ggplot(data = ps_df_sum3, aes(x = Sample, y = relative_abundance, fill = Genus)) +
  geom_bar(stat = "identity")+ scale_fill_manual(values= speciesPalette)+ scale_y_continuous(expand = c(0,0))+ylab("relative abundance")
p<-p+ theme(legend.text=element_text(size=10.5, face = "italic"))+theme(legend.key = element_rect(color = NA, fill = NA),legend.key.size = unit(0.5, "cm"))+theme(legend.title = element_blank())+theme(legend.position="bottom")
taxa2<-p+theme(axis.title.y = element_text(size=11, face="bold"))+theme(axis.text.y = element_text(size=10, face="bold"))+theme(axis.text.x = element_text(size=7, face="bold",vjust=0.5,angle = 90))+theme(axis.title.x = element_blank())+theme(strip.background =element_rect(fill="white", color="white"))+theme(strip.text = element_text(size=11, face="bold",vjust = -5))
taxa2<-taxa2+guides(fill=guide_legend(nrow=3))
taxa2<-taxa2+facet_grid(~source+treatment, scale='free_x', space = "free_x")

taxa_larva_adults<-taxa2+theme(axis.text.x = element_blank(), axis.ticks.x=element_blank())
#taxa2+facet_grid(~treatment, scale='free_x', space = "free_x")
ggsave("microbiome/combined_figure/merge_taxa.png", height = 4, width = 8)
```





# qPCR plots and statistics

#### 16S copy number and Actin as housekeeping control genes were quantified in adult and larva samples

```{r qpcr_setup}
library(rstatix)
library(broom)
library(olsrr)
```

```{r adults_load}

adults_qPCR<- read.table("qPCR/data_files/adults_qPCR.txt", header=TRUE)
all_bee<-subset(adults_qPCR, treatment != "Nurse")
all_bee <- droplevels(all_bee)
all_bee$treatment = factor(all_bee$treatment,levels=c("Hive","C","AG","LG","BB","LGBB"))

###16S copy number per bee

p<-ggerrorplot(all_bee, x = "treatment", y = "copies_16S_per_bee",desc_stat = "mean_se",error.plot = "errorbar",add ="mean",size=1.1,width=0.4,color = "gray35")+theme(legend.position = "none")+theme(legend.position = "none")+theme(axis.title.x = element_blank())+ ylab("16S copies per bee")
p<-p+ theme(axis.text.x = element_text(face="bold", size=15))+ theme(axis.text.y = element_text(face="bold", size=12))+theme(axis.title.y = element_text(face="bold", size=15))

p<-p+ geom_jitter(position=position_jitter(0.1), alpha=0.7,size=2.1,aes(colour=treatment))
p+ scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"))
ggsave("qPCR/plots/adults_total_16S_select_per_bee.png", height = 3.5, width = 6.5)
```

#### Pairwise comparisons
```{r stat}
AG<-subset(adults_qPCR, treatment == "AG")
BB<-subset(adults_qPCR, treatment == "BB")
C<-subset(adults_qPCR, treatment == "C")
LG<-subset(adults_qPCR, treatment == "LG")
LGBB<-subset(adults_qPCR, treatment == "LGBB")
Hive<-subset(adults_qPCR, treatment == "Hive")

wilcox.test(AG$copies_16S_per_bee,Hive$copies_16S_per_bee)
wilcox.test(BB$copies_16S_per_bee,Hive$copies_16S_per_bee)
wilcox.test(C$copies_16S_per_bee,Hive$copies_16S_per_bee)
wilcox.test(LG$copies_16S_per_bee,Hive$copies_16S_per_bee)
wilcox.test(LGBB$copies_16S_per_bee,Hive$copies_16S_per_bee)

pvalues <- c(1,0.11,0.45,0.04,0.45)
p.adjust(pvalues,method="fdr")
#1.0000 0.2750 0.5625 0.2000 0.5625
```

```{r ancova_adult}

ggplot(all_bee, aes(y = copies_16S_per_bee, x = Actin_copies_per_bee)) + geom_point() + geom_smooth(method = "lm")

model = lm(copies_16S_per_bee ~ Actin_copies_per_bee+treatment,data = all_bee)

#code to create the density plot of residuals
plot(density(model$residuals))

#code to create a Q-Q plot
qqnorm(model$residuals)
qqline(model$residuals, datax = FALSE, distribution = qnorm, probs = c(0.25, 0.75))

ols_test_normality(model)

#code to create a fitted values vs residuals plot
plot(model$residuals~model$fitted.values)
lines(lowess(model$fitted.values,model$residuals), col="blue")
text(model$fitted.values, model$residuals, row.names(all_bee), cex=0.6, pos=4, col="red")

# we should transform data
log_16S = log(all_bee$copies_16S_per_bee)

### set Hive control as reference treatment
levels(all_bee$treatment)

all_bee$treatment <- relevel(all_bee$treatment, ref="Hive")
levels(all_bee$treatment)

#code to run the ancova test on the transformed data
ancova<-lm(log_16S~Actin_copies_per_bee+treatment, data=all_bee)

#code to create the density plot of residuals
plot(density(ancova$residuals))

#code to create a Q-Q plot
qqnorm(ancova$residuals)
qqline(ancova$residuals, datax = FALSE, distribution = qnorm, probs = c(0.25, 0.75))

#code to create a fitted values vs. residuals plot
plot(ancova$residuals~ancova$fitted.values)
lines(lowess(ancova$fitted.values,ancova$residuals), col="blue")
text(ancova$fitted.values, ancova$residuals, row.names(all_bee), cex=0.6, pos=4, col="red")

#code to get the anova table
Anova(ancova,type="3")
```


```{r emmeans}

model.metrics <- augment(ancova) %>%
  select(-.hat, -.sigma, -.fitted) # Remove details
head(model.metrics, 10)

shapiro_test(model.metrics$.resid)

model.metrics %>% levene_test(.resid ~ treatment)
model.metrics %>% 
  filter(abs(.std.resid) > 3) %>%
  as.data.frame()

res.aov <- all_bee %>% anova_test(log(copies_16S)~Actin_copies_per_bee+treatment)
get_anova_table(res.aov)

# Pairwise comparisons
pwc <- all_bee %>% mutate(log16S = log(copies_16S)) %>%
  emmeans_test(
    log16S~treatment, covariate = Actin_copies_per_bee,
    p.adjust.method = "fdr")
pwc

pwc <- pwc %>% add_xy_position(x = "treatment", fun = "mean_se")

adult_qpcr<-ggline(get_emmeans(pwc), x = "treatment", y = "emmean", plot_type = c("p"), color = "treatment", size = 3) + geom_errorbar(aes(ymin = conf.low, ymax = conf.high), width = 0.4,size=2, colour="gray40", alpha=0.4)+stat_pvalue_manual(pwc, hide.ns = TRUE, tip.length = FALSE) +labs(subtitle = get_test_label(res.aov, detailed = TRUE),caption = get_pwc_label(pwc))+theme_classic()+theme(legend.position = "none")+
theme(axis.title.x = element_blank())+theme(axis.title.y = element_text(size=15, face="bold", colour="gray40"))+theme(axis.text.y = element_text(size=13))+theme(axis.text.x = element_text(size=15, face="bold"))+coord_cartesian(ylim = c(10, 14))+ scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"))
ggsave("qPCR/plots/emmeans_adults.png", height = 3.5, width = 6.5)

#pairwise comparisons against Hive control
#treatment order of plot
pvalues <- c(0.27,0.67,0.11,0.03,0.38)
p.adjust(pvalues,method="fdr")
#0.450 0.670 0.275 0.150 0.475
```


## Larva qPCR

### 16S copies/sample 
```{r larva}

all_larva<- read.table("qPCR/data_files/qPCR_data_for_R_larva.txt", header=TRUE)

all_larva$date = factor(all_larva$date,levels=c("start","day_three","day_six"), labels=c("start","day three","day six"))

## per sample
all_three<-subset(all_larva, date == "day three")
all_three$treatment = factor(all_three$treatment,levels=c("Hive","C","AG","LG","BB","LGBB"))

p1<-ggerrorplot(all_three,x = "treatment", y = "copies_16S_per_sample",desc_stat = "mean_se",error.plot = "errorbar",add ="mean",size=1.1,width=0.4,color = "gray35")+theme(legend.position = "none")+theme(legend.position = "none")+theme(axis.title.x = element_blank())+ ylab("16S copies per sample")+ylim(0, 6000000)
p1<-p1+ theme(axis.text.x = element_text(face="bold", size=15))+ theme(axis.text.y = element_text(face="bold", size=12))+theme(axis.title.y = element_text(face="bold", size=14))
p1<-p1+ geom_jitter(position=position_jitter(0.1), alpha=0.7,size=2.1,aes(colour=treatment))
p1+ scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"))
ggsave("qPCR/plots/larva_16S_day_three_per_sample.png", height = 4, width =6)

all_six<-subset(all_larva, date == "day six")
all_six$treatment = factor(all_six$treatment,levels=c("Hive","C","AG","LG","BB","LGBB"))

p2<-ggerrorplot(all_six,x = "treatment", y = "copies_16S_per_sample",desc_stat = "mean_se",error.plot = "errorbar",add ="mean",size=1.1,width=0.4,color = "gray35")+theme(legend.position = "none")+theme(legend.position = "none")+theme(axis.title.x = element_blank())+ ylab("Total 16S copies per sample")+ylim(0, 600000000)
p2<-p2+ theme(axis.text.x = element_text(face="bold", size=15))+ theme(axis.text.y = element_text(face="bold", size=12))+theme(axis.title.y = element_blank())
p2<-p2+ geom_jitter(position=position_jitter(0.1), alpha=0.7,size=2.1,aes(colour=treatment))
p2+ scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"))
ggsave("qPCR/plots/larva_total_16S_day_six_per_sample.png", height = 4, width = 6)

library(egg) #helps to correctly align the plots

figure_1 <- "qPCR/plots/combined_abundance_per_sample_larva.png"
png(figure_1, 11 * plot_res, 3 * plot_res, res = plot_res)
ggarrange(p1,p2, nrow=1,labels = c("A", "B"))
invisible(dev.off())
knitr::include_graphics(figure_1, dpi = plot_res)
```

#### Pairwise comparisons
```{r stat2}

AG<-subset(all_three, treatment == "AG")
BB<-subset(all_three, treatment == "BB")
C<-subset(all_three, treatment == "C")
LG<-subset(all_three, treatment == "LG")
LGBB<-subset(all_three, treatment == "LGBB")
Hive<-subset(all_three, treatment == "Hive")

wilcox.test(AG$copies_16S_per_sample,Hive$copies_16S_per_sample)
wilcox.test(BB$copies_16S_per_sample,Hive$copies_16S_per_sample)
wilcox.test(C$copies_16S_per_sample,Hive$copies_16S_per_sample)
wilcox.test(LG$copies_16S_per_sample,Hive$copies_16S_per_sample)
wilcox.test(LGBB$copies_16S_per_sample,Hive$copies_16S_per_sample)

pvalues <- c(0.008,0.22,0.15,0.008,0.008)
p.adjust(pvalues,method="fdr")
#0.01333333 0.22000000 0.18750000 0.01333333 0.01333333

AG2<-subset(all_six, treatment == "AG")
BB2<-subset(all_six, treatment == "BB")
C2<-subset(all_six, treatment == "C")
LG2<-subset(all_six, treatment == "LG")
LGBB2<-subset(all_six, treatment == "LGBB")
Hive2<-subset(all_six, treatment == "Hive")

wilcox.test(AG2$copies_16S_per_sample,Hive2$copies_16S_per_sample)
wilcox.test(BB2$copies_16S_per_sample,Hive2$copies_16S_per_sample)
wilcox.test(C2$copies_16S_per_sample,Hive2$copies_16S_per_sample)
wilcox.test(LG2$copies_16S_per_sample,Hive2$copies_16S_per_sample)
wilcox.test(LGBB2$copies_16S_per_sample,Hive2$copies_16S_per_sample)

pvalues <- c(0.61,0.002,0.004,0.18,0.002)
p.adjust(pvalues,method="fdr")
#0.610000000 0.005000000 0.006666667 0.225000000 0.005000000
```

```{r ancova_larva}

ggplot(all_six, aes(y = copies_16S_per_sample, x = Actin_copies_per_sample)) + geom_point() + geom_smooth(method = "lm")

model2 = lm(copies_16S_per_sample ~ Actin_copies_per_sample+treatment,data = all_six)

#code to create the density plot of residuals
plot(density(model2$residuals))

#code to create a Q-Q plot
qqnorm(model2$residuals)
qqline(model2$residuals, datax = FALSE, distribution = qnorm, probs = c(0.25, 0.75))

ols_test_normality(model2)

#code to create a fitted values vs residuals plot
plot(model2$residuals~model2$fitted.values)
lines(lowess(model2$fitted.values,model2$residuals), col="blue")
text(model2$fitted.values, model2$residuals, row.names(all_six), cex=0.6, pos=4, col="red")

# we should transform data
log_16S = log(all_six$copies_16S_per_sample)

### set Hive control as reference treatment
levels(all_six$treatment)

all_six$treatment <- relevel(all_six$treatment, ref="Hive")
levels(all_six$treatment)

#code to run the ancova test on the transformed data
ancova2<-lm(log_16S~Actin_copies_per_sample+treatment, data=all_six)

#code to create the density plot of residuals
plot(density(ancova2$residuals))

#code to create a Q-Q plot
qqnorm(ancova2$residuals)
qqline(ancova2$residuals, datax = FALSE, distribution = qnorm, probs = c(0.25, 0.75))

#code to create a fitted values vs. residuals plot
plot(ancova2$residuals~ancova2$fitted.values)
lines(lowess(ancova2$fitted.values,ancova2$residuals), col="blue")
text(ancova2$fitted.values, ancova2$residuals, row.names(all_six), cex=0.6, pos=4, col="red")

#code to get the anova table
Anova(ancova2,type="3")
```

```{r emmeans_day_six}
model.metrics <- augment(ancova2) %>%
  select(-.hat, -.sigma, -.fitted) # Remove details
head(model.metrics, 10)

shapiro_test(model.metrics$.resid)

model.metrics %>% levene_test(.resid ~ treatment)
model.metrics %>% 
  filter(abs(.std.resid) > 3) %>%
  as.data.frame()

res.aov <- all_six %>% anova_test(log(copies_16S)~Actin_copies_per_sample+treatment)
get_anova_table(res.aov)

# Pairwise comparisons
pwc2 <- all_six %>% mutate(log16S = log(copies_16S)) %>%
  emmeans_test(
    log16S~treatment, covariate = Actin_copies_per_sample,
    p.adjust.method = "fdr")
pwc2

pwc2 <- pwc2 %>% add_xy_position(x = "treatment", fun = "mean_se")

larva_qpcr<-ggline(get_emmeans(pwc2), x = "treatment", y = "emmean", plot_type = c("p"), color = "treatment", size = 3) + geom_errorbar(aes(ymin = conf.low, ymax = conf.high), width = 0.4,size=2, colour="gray40", alpha=0.4)+stat_pvalue_manual(pwc, hide.ns = TRUE, tip.length = FALSE) +labs(subtitle = get_test_label(res.aov, detailed = TRUE),caption = get_pwc_label(pwc))+theme_classic()+theme(legend.position = "none")+
theme(axis.title.x = element_blank())+theme(axis.title.y = element_text(size=15, face="bold", colour="gray40"))+theme(axis.text.y = element_text(size=13))+theme(axis.text.x = element_text(size=15, face="bold"))+coord_cartesian(ylim = c(3, 14.5))+ scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"))
ggsave("qPCR/plots/emmeans_larva_day_six.png", height = 3.5, width = 6.5)

#pairwise comparisons against Hive control
#treatment order of plot
pvalues <- c(0.0000000000015,0.96,0.33,0.00000000037,0.019)
p.adjust(pvalues,method="fdr")
#7.500000e-12 0.96 0.4 9.250000e-10 0.03
```

```{r day_three}
## per sample
all_three<-subset(all_larva, date == "day three")
all_three$treatment = factor(all_three$treatment,levels=c("Hive","C","AG","LG","BB","LGBB"))
# we should transform data
log_16S = log(all_three$copies_16S_per_sample)

### set Hive control as reference treatment
levels(all_three$treatment)

all_three$treatment <- relevel(all_three$treatment, ref="Hive")
levels(all_three$treatment)

#code to run the ancova test on the transformed data
ancova2<-lm(log_16S~Actin_copies_per_sample+treatment, data=all_three)

#code to create the density plot of residuals
plot(density(ancova2$residuals))

#code to create a Q-Q plot
qqnorm(ancova2$residuals)
qqline(ancova2$residuals, datax = FALSE, distribution = qnorm, probs = c(0.25, 0.75))

#code to create a fitted values vs. residuals plot
plot(ancova2$residuals~ancova2$fitted.values)
lines(lowess(ancova2$fitted.values,ancova2$residuals), col="blue")
text(ancova2$fitted.values, ancova2$residuals, row.names(all_six), cex=0.6, pos=4, col="red")

#code to get the anova table
Anova(ancova2,type="3")
```

```{r emmeans_day_three}

# Inspect the model diagnostic metrics
model.metrics <- augment(ancova2) %>%
  select(-.hat, -.sigma, -.fitted) # Remove details
head(model.metrics, 10)

shapiro_test(model.metrics$.resid)

model.metrics %>% levene_test(.resid ~ treatment)
model.metrics %>% 
  filter(abs(.std.resid) > 3) %>%
  as.data.frame()

res.aov <- all_three %>% anova_test(log(copies_16S)~Actin_copies_per_sample+treatment)
get_anova_table(res.aov)

# Pairwise comparisons
pwc2 <- all_three %>% mutate(log16S = log(copies_16S)) %>%emmeans_test(
    log16S~treatment, covariate = Actin_copies_per_sample,
    p.adjust.method = "fdr")
pwc2
pwc2 <- pwc2 %>% add_xy_position(x = "treatment", fun = "mean_se")

ggline(get_emmeans(pwc2), x = "treatment", y = "emmean", plot_type = c("p"), color = "treatment", size = 3) + geom_errorbar(aes(ymin = conf.low, ymax = conf.high), width = 0.4,size=2, colour="gray40", alpha=0.4)+stat_pvalue_manual(pwc, hide.ns = TRUE, tip.length = FALSE) +labs(subtitle = get_test_label(res.aov, detailed = TRUE),caption = get_pwc_label(pwc))+theme_classic()+theme(legend.position = "none")+
  theme(axis.title.x = element_blank())+theme(axis.title.y = element_text(size=15, face="bold", colour="gray40"))+theme(axis.text.y = element_text(size=13))+theme(axis.text.x = element_text(size=15, face="bold"))+coord_cartesian(ylim = c(2.5, 9))+ scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79","cornflowerblue","plum"))
ggsave("qPCR/plots/emmeans_larva_day_three.png", height = 3.5, width = 6.5)

#pairwise comparisons against Hive control
#treatment order of plot
pvalues <- c(0.14,0.0000048,0.00000005,0.28,0.0000000034)
p.adjust(pvalues,method="fdr")
#0.175 8.00e-06 1.25e-07 0.28 1.70e-08
```


## DNA vs RNA comparison across some larva samples

```{r DNA_vs_RNA_larva}

compare2<- read.table("qPCR/data_files/qPCR_data_larva_RNA_and_DNA.txt", header=TRUE)
compare2$treatment <- factor(compare2$treatment, levels = c("Hive","C","BB","LG"))

p<-ggerrorplot(compare2,x = "treatment", y = "copies_16S_per_sample",desc_stat = "mean_se",error.plot = "errorbar",add ="mean",size=1.1,width=0.4,color = "type",position = position_dodge(0.4))+theme(legend.title = element_blank())+theme(axis.title.x = element_blank())+ ylab("Total 16S copies per sample")+scale_color_manual(values=c("#55596a", "#6ebe9f"),guide = guide_legend(override.aes = list(shape = c(rep(15, 2)),size=1.9)))+theme(legend.text = element_text(size=15, face="bold"))
p<-p+ theme(axis.text.x = element_text(face="bold", size=15))+ theme(axis.text.y = element_text(face="bold", size=12))+theme(axis.title.y = element_text(face="bold", size=14))
p<-p+ geom_jitter(position=position_jitter(0.2), alpha=0.7,size=2.1,aes(colour=type))+theme(legend.position='right')+ theme(legend.key.size = unit(1.5, 'lines'))
p
ggsave("qPCR/plots/DNA_RNA_larva_16S.png", height = 4, width = 7)

####log plot

p<-ggerrorplot(compare2,x = "treatment", y = "log_copies_16S_per_sample",desc_stat = "mean_se",error.plot = "errorbar",add ="mean",size=1.1,width=0.4,color = "type",position = position_dodge(0.4))+theme(legend.title = element_blank())+theme(axis.title.x = element_blank())+ ylab("Total 16S copies per sample")+scale_color_manual(values=c("#55596a", "#6ebe9f"),guide = guide_legend(override.aes = list(shape = c(rep(15, 2)),size=1.9)))+theme(legend.text = element_text(size=15, face="bold"))
p<-p+ theme(axis.text.x = element_text(face="bold", size=15))+ theme(axis.text.y = element_text(face="bold", size=12))+theme(axis.title.y = element_text(face="bold", size=14))
p<-p+ geom_jitter(position=position_jitter(0.2), alpha=0.7,size=2.1,aes(colour=type))+theme(legend.position='right')+ theme(legend.key.size = unit(1.5, 'lines'))
ggsave("qPCR/plots/DNA_RNA_larva_16S_log.png", height = 4, width = 7)
```

```{r statsDNA_vs_RNA_larva}
## compare RNA and DNA qPCR results for each treatment. copy number per sample are chosen

Hive_DNA_day_6<- c(158149260.6,299464933.4,38329650.55,210743448.4,157158956.8,29642113.99)
Hive_RNA_day_6<- c(592517966.1,69099913.77,6490223.3,5300421.329,2200292.495,761495.3369)
wilcox.test(Hive_DNA_day_6, Hive_RNA_day_6)
wilcox.test(Hive_DNA_day_6, Hive_RNA_day_6,paired = T)
wilcox.test(log(Hive_DNA_day_6), log(Hive_RNA_day_6),paired = T)

C_DNA_day_6<- c(136144.1971,327896.419,338489.8574,191852.3266,83310.3936)
C_RNA_day_6<- c(20576.18118,2939.033432,7064.685271,10102.5261,53527.54112)
wilcox.test(C_DNA_day_6, C_RNA_day_6,paired = T)
wilcox.test(log(C_DNA_day_6), log(C_RNA_day_6),paired = T)

BB_DNA_day_6<- c(365559.5144,727079.2007,3817659.064,3623182.624,2676524.643,895496.4022)
BB_RNA_day_6<- c(118855.4821,25123.08346,202746.5693,214357.3069,1335941.49,116913.895)
wilcox.test(BB_DNA_day_6, BB_RNA_day_6)
wilcox.test(BB_DNA_day_6, BB_RNA_day_6,paired = T)
wilcox.test(log(BB_DNA_day_6), log(BB_RNA_day_6),paired = T)

LG_DNA_day_6<- c(251392143.2,241259678.6,87163959.39,532901560.6,260541064.5,196896870.5)
LG_RNA_day_6<- c(159421944.5,9352876.355,1151030.41,113017.632,2050368.6,1780547.052)
wilcox.test(LG_DNA_day_6, LG_RNA_day_6)
wilcox.test(LG_DNA_day_6, LG_RNA_day_6,paired = T)
wilcox.test(log(LG_DNA_day_6), log(LG_RNA_day_6),paired = T)
```





# combined figure creation

```{r, figure_1}

legend <- get_legend(PCoA_adults)
legend<-as_ggplot(legend)
ggsave("microbiome/combined_figure/legend.png", height = 1, width = 11)

legend2 <- get_legend(taxa_larva_adults)
legend2<-as_ggplot(legend2)
ggsave("microbiome/combined_figure/legend2.png", height = 2, width = 11)

Observed2<-Observed_adults+theme(legend.position = "none")
PCoA2<-PCoA_adults+theme(legend.position = "none")
Observed3<-Observed_larvae+theme(legend.position = "none")
PCoA3<-PCoA_larvae+theme(legend.position = "none")
larvae_qPCR<-larva_qpcr+theme(legend.position = "none")
adult_qPCR<-adult_qpcr+theme(legend.position = "none")

library(egg) #helps to correctly align the plots

figure_1 <- "microbiome/combined_figure/manuscript_main_all_new.png"
png(figure_1, 11 * plot_res, 9 * plot_res, res = plot_res)
ggarrange(PCoA3,PCoA2,Observed3,Observed2,larvae_qPCR,adult_qPCR,nrow=3) 
invisible(dev.off())
knitr::include_graphics(figure_1, dpi = plot_res)
```


